RESEARCH ARTICLE

Cluster Analysis According to Immunohistochemistry is a
Robust Tool for NonûSmall Cell Lung Cancer and Reveals
a Distinct, Immune Signature-defined Subgroup
William Sterlacci, MD,* Michael Fiegl, MD,åç Darius Juskevicius, PhD,º
and Alexandar Tzankov, MDº

Abstract: Clustering in medicine is the subgrouping of a cohort
according to speci?c phenotypical or genotypical traits. For breast
cancer and lymphomas, clustering by gene expression pro?les has
already resulted in important prognostic and predictive subgroups.
For nonûsmall cell lung cancer (NSCLC), however, little is known.
We performed a cluster analysis on a cohort of 365 surgically re-
sected, well-documented NSCLC patients, which was followed-up
for a median of 62 months, incorporating 70 expressed proteins and
several genes. Our data reveal that tumor grading by architecture is
signi?cant, that large cell carcinoma is likely not a separate entity,
and that an immune signature cluster exists. For squamous cell
carcinomas, a prognostically relevant cluster with poorer outcome
was found, de?ned by a high CD4/CD8 ratio and lower presence of
granzyme B+ tumor-in?ltrating lymphocytes (TIL). This study
shows that clustering analysis is a useful tool for verifying established
characteristics and generating new insights for NSCLC. Im-
portantly, for one ôimmune signatureö cluster, the signature of the
TIL (especially the amount of CD8+ TIL) was more crucial than the
histologic or any other phenotypical aspect. This may be an im-
portant ?nding toward explaining why only a fraction of eligible
patients respond to immunomodulating anticancer therapies.

Key Words: nonûsmall cell
immunohistochemistry

lung cancer, cluster analysis,

(Appl Immunohistochem Mol Morphol 2019;00:000û000)

I n medicine, clustering is known as the grouping of ob-

jects by calculating similarities; thus, objects in one
cluster are more closely related than those found in other
clusters. Especially in association with high-throughput
protein and gene analyses, the utilization of clustering has

Received for publication September 24, 2018; accepted January 30, 2019.
From the *Institute of Pathology, Hospital Bayreuth, Preuschwitzerstrasse,
Bayreuth, Germany; åDepartment of Internal Medicine, Division
of Hematology and Oncology, Medical University Innsbruck, Anich-
strasse, Innsbruck; çDepartment of Internal Medicine, Oncology and
Geriatrics, Hospital Hochrum, Lõrchenstrasse, Rum, Austria; and
ºInstitute of Pathology, University Hospital Basel, Schoenbeinstrasse,
Basel, Switzerland.

The authors declare no con?ict of interest.
Reprints: William Sterlacci, MD, Institute of Pathology, Hospital Bayr-
euth, Preuschwitzerstrasse 101, Bayreuth 95445, Germany (e-mail:
william.sterlacci@klinikum-bayreuth.de).

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

tool

for discovering different

become increasingly popular for characterizing different
subgroups of cancer. Particularly in breast cancer and in
lymphomas, cluster analysis has been reported as a pow-
classes.1,2 Sub-
erful
classi?cation of heterogenous cohorts makes it possible to
identify distinct prognostic or predictive subgroups that
are, for example, potentially linked to speci?c treatment
outcomes. In breast cancer, gene expression analyses
based on clustering have made it possible to distribute
morphologically similar tumors into groups showing dif-
ferent prognosis as well as different treatment responses.3,4
In the future, such expression pro?les will likely also be
incorporated into the tumor classi?cation systems.

The proteins that are present in tumor cells are de?ned
by individual genetic programming. On the one hand, this
programming underlies intrinsic tumor-associated factors, such
as mutations (point mutations and copy number variations),
especially of genes encoding transcription factors or genes in-
volved in signaling cascades [eg, anaplastic lymphoma kinase
(ALK), epidermal growth factor receptor (EGFR), hepatocyte
growth factor receptor (MET), and RAS-family members]. In
contrast, extrinsic factors are also in?uential, including inter-
action with immune cells, for example, programmed death 1ù
programmed death ligand 1 (PD1-PDL1) or tumor-in?ltrating
?broblasts. Usually not the expression of a single protein, but
rather of a whole group of proteins (ie, cluster) is involved.
Using mass spectrometry, it would be possible to analyze the
entire protein expression pro?le of a tumor sample; however,
this method cannot differentiate the exact origin of these
proteins [ie, tumor cells, tumor-in?ltrating lymphocytes (TIL),
macrophages, ?broblasts, endothelial cells, etc.].

Gene expression pro?ling data in nonûsmall cell
lung cancer (NSCLC) also point toward the possibility of
subclassi?cation and prognostic importance; however, this
issue does not yet seem as relevant compared with breast
cancer.5 Nevertheless, pro?ling data of NSCLC con-
tributed to pathogenesis understanding, showed the pos-
sibility of further subgrouping of the histologic types, and
can predict disease-free and overall survival.6,7

Importantly, clustering analyses are also possible
with immunohistochemical expression pro?les and have
been performed in several tumors including lymphomas
and breast cancer.1,8,9 For NSCLC, however, there are
only a few reports on small cohorts with few analyzed
parameters, or limited to a single stage.10,11

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

www.appliedimmunohist.com | 1

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Sterlacci et al

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

TABLE 1. Antibodies Applied and Cut-off Scores
Antibodies

Source and Clone or ID

ABCG5
ALDH1
ALK
Ber-Ep4
Cbl-b
CD4
CD8
CD34
CD24
CD44
CD44v6
CD56 (NCAM)
CD95
CD105
CD166
Chromogranin A
CXCR4
CXCR4 phosphorylated
Cyclin D1
Cyclin D2
Cyclin D3
Cyclin E
Cytokeratin 5/6
Cytokeratin 7
Cytokeratin 34?E12
D2-40
E-cadherin
EGFR
ESA
FoxP3
Granzyme B
Ki67
Mast cell tryptase
Mel-CAM
MET
Moc31
Mucin1
Mucin2
Mucin4
Mucin5AC
Mucin6
Nestin
Neuron-speci?c enolase
OCT4
OPN
p16
p21
p27
p63
pAkt
PD1
PDL1
Pgp
PGP 9.5
PI3K
pSTAT3
PTEN
RHAMM
SDF-1
Snail
SOX2
Synaptophysin
TGF-?
TIA1
TTF1
VE-cadherin
VEGF
VEGFR1

Sigma HPA016514
Abcam EP1933Y
Ventana ALK01
Ventana Ber-EP4
Abcam 246C5a
Cell Marque SP35
DAKO C8/144B
Ventana QBEnd/10
ThermoFisher SN3b
Ventana SP37
Medical Systems ABIN 123880
Ventana MRQ-42
Novocastra GM30
Novocastra 4G11
Novocastra MOG/07
Neomarkers MS-382-P
Abcam ab2074
Abcam ab74012
ThermoFisher DCS-6
Santa Cruz sc-181
Novocastra DCS-22
Neomarkers 13A3
Ventana D5/16B4
Progen Ks7.18
Ventana 34bE12
Ventana D2-40
Ventana EP700Y
Ventana 3C6
Novocastra VU-1D9
Abcam mAbcam 22510
Novocastra 11F1
DAKO MIB-1
DAKO AA1
Novocastra N1238
Ventana SP44
DAKO MOC-31
Ventana H23
Ventana MRQ-18
Biocare 8G-7
Ventana MRQ-19
Ventana MRQ-20
AbD Serotec 10C2
Ventana MRQ-55
Ventana MRQ-10
Novocastra OP3N
CINtec E6H4
ThermoFisher HZ52
Neomarkers DCS-72.F6
Ventana 4A4
Abcam ab8932
Cell Marque NAT105
Cell signaling E1L3N
Enzo Lifesciences P3II-26
Novocastra 10A1
BD Bioscience 4/PI3-Kinase
Cell Signaling D3A7
Cell signaling 138G6
Novocastra 2D6
Abcam ab135949
Abgent AP2054a
Abcam EPR3131
Novocastra 27G12
Acris DM1047
Immunotech 2G9A10F5
Neomarkers 8G7G3/1
Novocastra 3E1
DAKO VG1
Neomarkers RB-1527-P0

Dilution

1:200*
1:200
Ready to use
Ready to use
1:50
1:100
1:400
Ready to use
1:20
Ready to use
1:100
Ready to use
1:400
1:80*
1:200*
1:400
1:50
1:100
1:50
1:8000
1:80
1:20
Ready to use
1:50å
Ready to use
Ready to use
Ready to use
Ready to useå
1:800
1:50
1:100
1:100
1:100
1:25
Ready to use
Ready to use
Ready to use
Ready to use
1100
Ready to use
Ready to use
1:200
Ready to use
Ready to use
1:50
Ready to use
1:400
1:250
Ready to use
1:450
1:50
1:50
1:4000*
1:20
1:1000*
1:50*
1:200
1:25*
1:20
1:50*
1:100*
1:100
1:50
1:100
1:25
1:100
1:40*
1:20*

Cut-off Score

Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
> 19% positive tumor cells
> 3 TILs/mm2
> 53 TILs/mm2
> 35 TILs/mm2
Any expression in tumor cells
> 9% positive tumor cells
Any expression in tumor cells
> 8% positive tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
> 9% positive tumor cells
Any expression in tumor cells
> 1.5% positive tumor cells
> 3% positive tumor cells
> 15% positive tumor cells
Any expression in tumor cells
> 3.5% positive tumor cells
> 1% positive tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any TILs
Any TILs
> 3%
> 10 TI mast cells/mm2
Any expression in tumor cells
> 49% positive tumor cells
> 49% positive tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
> 3.5% positive tumor cells
> 9% positive tumor cells
> 15% positive tumor cells
> 45% nuclear positive tumor cells
Any expression in tumor cells
Any expression in tumor cells
> 14 TILs/mm2
> 4% positive tumor cells
> 0.5% positive tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
< 10% positive tumor cells
> 4% positive tumor cells
> 1.4% positive tumor cells
Any expression in tumor cells
> 2% positive tumor cells
Any expression in tumor cells
> 3 TILs/mm2, any expression in tumor cells
> 7 TILs/mm2
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells
Any expression in tumor cells

2 | www.appliedimmunohist.com

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

Cluster Analysis According to Immunohistochemistry

TABLE 1. (continued)
Antibodies

VEGFR2
?-catenin

Source and Clone or ID

Neomarkers RB-10453-P1
Ventana 14

Dilution

1:10*
Ready to use

Cut-off Score

Any expression in tumor cells
Nuclear expression in tumor cells

In all instances except for * and å, in which high pH buffers or protease have been applied, respectively, antigen retrieval was based on lower pH buffers and microwaving.
ABCG5 indicates ATP binding cassette subfamily G member 5; ALDH1, aldehyde dehydrogenase 1; ALK, anaplastic lymphoma kinase; Cbl-b, casitas B-lineage lymphoma
(protooncogene) B; CXCR4, chemokine receptor; EGFR, epidermal growth factor; ESA (Ber-EP4, MOC31), epithelial-speci?c antigen; FoxP3, forkhead-box protein 3; MET, hepatocyte
growth factor receptor; OCT4, octamer binding transcription factor 4; OPN, osteopontine; PD1, programmed death (receptor) 1; PDL1, programmed death ligand 1; Pgp, P-glycoprotein;
PGP 9.5, protein gene product 9.5; PI3K, phosphoinositide 3-kinase; pSTAT3, phosphorylated signal transducer and activator of transcription 3; PTEN, phosphatase and tensine homolog;
RHAMM, receptor of hyaluronic acid mediated motility; SOX2, sex-determining region Y-box 2; TGF-?, transforming growth factor beta; TIA1, cytotoxic granule-associated RNA
binding protein; TIL, tumor-in?ltrating lymphocytes; TTF1, thyroid transcription factor 1; VEGF, vascular endothelial growth factor; VEGFR1, VEGF receptor 1.

Over the past years, we have analyzed over 70 pro-
teins and protein combinations (Table 1), the gene status
(by means of in situ hybridization) of ALK, cyclin D 1
(CCND1), EGFR, and MET, and the methylation status
of p16 of 405 clinically well-characterized NSCLC patients
who underwent surgical resection with curative intention
and who were followed-up over a median of 62 months
(range, 0.1 to 223 mo) as a part of the retrospective Twenty
Years Retrospective of Lung Cancer (TYROL) study.12
Formalin-?xed and paraf?n-embedded tumor tissue material
of these patients was brought into a tissue microarray (TMA)
platform,13 and the number of analyzable cases decreased

slightly from 405 to 365 over time due to TMA exhaustion.
Besides markers associated with certain histologic subtypes,
parameters crucial for neuroendocrine differentiation, cell
cycle regulation, apoptosis, cell motility and adhesion, the
immunologic microenvironment, stemness, and tumor micro-
vasculature were analyzed. Certain predictive markers, for
example, expression of ALK, PDL1, and MET, were also
assessed.13û26

The aim of the present study was to apply clustering
analysisùa potentially powerful tool, which has rarely
been utilized for NSCLC so farùto all previously assessed
parameters for further comprehensive characterization

FIGURE 1. Heatmap of unsupervised hierarchical clustering analysis of the studied 365 nonûsmall cell lung cancers. The upper
dendrogram shows the patients split into 3 clusters, which are separated by red vertical lines. The left-handed dendrogram reflects
marker clustering. For markers, yellow indicates lacking (below cut-off) expression/negativity or CD4/CD8 ratio <1, while green
indicates expression/positivity or CD4/CD8 ratio > 1. Empty spots depict markers that were not analyzable in respective cases. Only
significant (P < 0.000055) discriminating markers or marker groups as well as some other informative markers (grade > 1, Ki67,
neuroendocrine markers) are annotated. The 3-colored line between the upper dendrogram and the heatmap shows the corre-
sponding histopathologic classification of cases: yellowùACA, redùSCC, and orangeùLCC. Note that ACA mostly cluster to cluster 1
(ACA-like) and SCC to cluster 2 (SCC-like), while LCC are distributed throughout the unsupervised clusters 1, 2, and 3. Interestingly,
a third unsupervised cluster 3 (right-handed) mostly defined by TIL signatures and containing ACA, SCC, and LCC appears. ACA
indicates adenocarcinoma; ALDH1, aldehyde dehydrogenase 1; Ep-CAM, epithelial cell adhesion molecule; LCC, large cell carcinoma;
SCC, squamous cell carcinoma; SOX2, sex-determining region Y-box 2; TIL, tumor-infiltrating lymphocyte.

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

www.appliedimmunohist.com | 3

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Sterlacci et al

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

and subgrouping of our cohort
to reveal additional
biological and prognostic strati?cations. This represents
the largest NSCLC collective to date analyzed by protein
expression clustering analysis, and incorporates the widest
panel of markers, providing a ?rst thorough look for this
technique in surgically resected NSCLC.

METHODS

Cases

The archival samples derived from NSCLC patients
with radical surgical resection with curative intent between
1992 and 2004 and diagnosed at the Institute of Pathology,
Medical University of Innsbruck, were studied.13 The cohort
reported here consists of 365 cases. Carcinoids were excluded
from this analysis. Cases were selected only on the basis of
tissue preservation. Hematoxylin and eosin (H&E)-stained
slides from all available specimens were reclassi?ed by 2
pathologists (W.S. and A.T.) without knowledge of patient
data, according to the current (2015) WHO classi?cation of
tumors of the lung.27 Tumor differentiation was graded as
well, as moderate or poor. The clinical information (including
parameters such as tumor stage, disease recurrence, and
overall survival) was documented within the TYROL survey,
a project aiming to analyze various features of a large
number of lung cancer patients.12 Approval for data ac-
quisition and analysis was obtained from the Ethics Com-
mittee of the Medical University of Innsbruck.

TMA Construction

The tumor material consisted of paraf?n-embedded
tissue after ?xation in 10% neutral buffered formalin. The
TMA was constructed, as previously described.13 The ?rst
sections were stained by H&E, Periodic Acid-Schiff (PAS),
and alcian-blue-PAS to con?rm validity; the following
were used for immunohistochemistry.

Phenotypic and Genotypic Marker Analyses

Starting in the year 200913 and ending in the year
2018,26 over 70 single-protein expression analyses have

TABLE 2. Unsupervised Cluster Analysis

been performed by means of immunohistochemistry, and
4 NSCLC-relevant genes, that is, ALK, CCND1, EGFR,
and MET, have been studied by means of interphase ?u-
orescence in situ hybridization (FISH) in that cohort,13û26
and clinical follow-up has been updated until 2016. Im-
portantly, for all markers rational, diagnostically useful,
prognostically relevant, or biologically meaningful cut-off
scores have been applied; Table 1 incorporates all antibodies
and FISH-probes applied, pretreatment conditions, cut-off
scores used, and number of positive cases.

Statistical Analysis

Hierarchical cluster analysis was used to organize
markers and cases, respectively, according to their sim-
ilarities. Immunohistochemical marker expression and FISH
data were converted to binary format representing positivity
or negativity based on the prede?ned cut-offs (Table 1).
Distance matrix was calculated using the JaccardÆs distance
measure, and hierarchical clustering with complete linkage
was performed applying R statistical software (version 3.3.0,
https://www.r-project.org/).8,9,28 Analysis was performed on
the entire data set row-wise (cases) and column-wise (mark-
ers) and visualized as a binary heatmap with adjunct den-
drograms. To evaluate the stability of selected clusters, that is,
reproducibility of clustering, NSCLC cases were randomly
split into 2 groups and reclustered. Allocation of randomly
selected cases into the initially obtained clusters was evaluated
by the Cohen ?, ? > 0.61 implying substantial, > 0.41, mod-
erate, and > 0.21 fair reproducibility. Hierarchical clustering
was also performed, as described above in a supervised
manner on case subsets according to their histologic subtype,
that is, squamous cell carcinoma (SCC), adenocarcinoma
(ACA), and large cell carcinomas (LCC).
Distribution analysis of

im-
munohistochemical, and FISH parameters with regard to
clusters was performed using the ?2 test, corrected for
multiple testing with the Bonferroni method, considering
P ? 0.000055 as statistically signi?cant in the unsupervised
entire data set, and ? 0.000018 in the supervised cluster
analysis of separate histologic subtypes.

clinicopathologic,

Parameters

Cluster 1 (n = 206)

Cluster 2 (n = 105)

Cluster 3 (n = 54)

Increased

CK7
TTF1
TGF-?
EGFR
E-cadherin
Ber-EP4
MOC31
ESA
MUC5AC
p16
p27c
CD24

Decreased

CD24 û /CD44s+
ù
ù
ù
ù
ù
ù
ù
ù
ù
ù
ù

Increased

CK5/6
CK34?E12
p63
CD44s
CD44v6
CCNE
p21
ALDH1
SOX2
ù
ù
ù

Decreased

Increased

Decreased

ù
ù
ù
ù
ù
ù
ù
ù
ù
ù
ù
ù

CD8+ TIL
TGF-?+ TIL
ù
ù
ù
ù
ù
ù
ù
ù
ù
ù

CD4/CD8
Ber-EP4
MOC31
ESA
ù
ù
ù
ù
ù
ù
ù
ù

ALDH1 indicates aldehyde dehydrogenase 1; c, cytoplasmic; CCNE, cyclin E; CD4/CD8, CD4/CD8 ratio; ChrA, chromogranin A; CK7, cytokeratin 7; EGFR, epidermal
growth factor; ESA(Ber-EP4, MOC31), epithelial-speci?c antigen; MUC5AC, mucin 5AC; SOX2, sex-determining region Y-box 2; TGF-?, transforming growth factor beta;
TIL, tumor-in?ltrating lymphocytes; TTF1, thyroid transcription factor 1.

4 | www.appliedimmunohist.com

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

Cluster Analysis According to Immunohistochemistry

Kaplan-Meier curves were calculated for survival
estimates, and exploratory (ie, not corrected for multiple
testing) log-rank statistics was used to determine prog-
nostic differences between clusters. When not corrected
for multiple testing, P < 0.05 were considered as sig-
ni?cant. Two-sided tests were used throughout. Statistical
calculations with regard to Cohen ?, distribution, and
survival were performed using SPSS 22.0 software (SPSS,
Chicago, IL).

RESULTS AND DISCUSSION

Unsupervised cluster analysis revealed 3 NSCLC clus-
ters (Fig. 1 and Table 2): one consisting of 206 cases showing
higher expression of CK7, thyroid transcription factor 1
(TTF1), transforming growth factor beta (TGF-?), EGFR,
E-cadherin, and epithelial cell adhesion molecule (Ep-CAM)
compounds (Ber-EP4, MOC31 and ESA), MUC5AC, p16,
cytoplasmic p27, and CD24 and low presence of cases with
the stem cellûlike phenotype (CD24 û /CD44s+), and thus

FIGURE 2. Heatmap of hierarchical clustering analysis of the studied 218 ACA. Patients split into 3 clusters, which are separated by
vertical lines. For markers, light grey indicates lacking (below cut-off) expression/negativity, whereas dark grey indicates expression/
positivity. Only significant (P < 0.000018) discriminating markers or marker groups as well as some other informative markers
(grade > 1, CK5/6, CK34?E12, p63, ACA markers and neuroendocrine markers, stainable mucins) are annotated. The grayscale
colored heatmap between the upper dendrogram and the marker heatmap codes for relevant clinicopathologic data, black
indicating patients above 65 years old, male individuals, smokers, patients suffering from relapses, higher pathological nodal,
pathological tumor, and pathological Union Internationale Contre le Cancer stages or cases displaying lepidic growth patterns,
respectively. Empty spots depict markers that were not analyzable in respective cases or in cases with missing clinicopathologic
data. ACA indicates adenocarcinoma; ALDH1, aldehyde dehydrogenase 1; EGFR, epidermal growth factor; Ep-CAM, epithelial cell
adhesion molecule; SOX2, sex-determining region Y-box 2; TIL, tumor-infiltrating lymphocyte; TTF1, thyroid transcription factor 1.

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

www.appliedimmunohist.com | 5

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Sterlacci et al

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

FIGURE 3. Heatmap of hierarchical clustering analysis of the studied 125 SCC. Patients split into 3 clusters, which are separated by vertical
lines. For markers, light grey indicates lacking (below cut-off) expression/negativity or CD4/CD8 ratio <1, while dark grey indicates expression/
positivity or CD4/CD8 ratio > 1. Only significant (P < 0.000018) discriminating markers or marker groups as well as some other informative
markers (grade > 1, CK7, TTF1, Ki67, SCC-markers, subpopulations of TIL) are annotated. The grayscale colored heatmap between the upper
dendrogram and the marker heatmap codes for relevant clinicopathologic data, black indicating patients above 65 years old, male in-
dividuals, smokers, patients suffering from relapses, higher pathological nodal, pathological tumor, and pathological Union Internationale
Contre le Cancer stages, or cases displaying keratinization, respectively. Empty spots depict markers that were not analyzable in respective
cases or cases with missing clinicopathologic data. CK7 indicates cytokeratin 7; Ep-CAM, epithelial cell adhesion molecule; SCC indicates
squamous cell carcinoma; TIL, tumor-infiltrating lymphocyte; TTF1, thyroid transcription factor 1.

6 | www.appliedimmunohist.com

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

Cluster Analysis According to Immunohistochemistry

displaying an ACA-like phenotype (unsupervised cluster 1);
another consisting of 105 cases with higher expression
of CK5/6, CK34?E12, p63, CD44s and CD44v6, cyclin
E (CCNE) and p21, as well as aldehyde dehydrogenase
1 (ALDH1) and sex-determining region Y-box 2 (SOX2),
and thus displaying an SCC-like phenotype (unsupervised
cluster 2); and a third cluster of 54 cases with higher intra-
tumoral contents of CD8-positive T cells, lower CD4/CD8
ratios, higher amounts of TGF-?-positive lymphocytes, and
lower expression of Ep-CAM compounds (unsupervised
ôimmune signatureö cluster 3). P-values for the linkage of all

the above markers to the respective clusters were <0.000055.
As to be expected, skewing toward male sex was observable
in the unsupervised cluster 2 (SCC-like), while neuro-
endocrine markers were more commonly present in the un-
supervised clusters 1 and 3.
Reanalysis of

the randomly dichotomized entire
NSCLC collective to test reproducibility of clustering
yielded a ?-value of 0.733 for the ACA-like cluster 1 and
0.692 for the SSC-like cluster 2, indicating substantial re-
producibility, whereas the smaller cluster 3 was only fairly
reproducible with a ?-value of 0.371.

Thus, the unsupervised cluster analysis of this large
collective of NSCLC cases demonstrates the very robust
and substantially reproducible signature of the biological
diversity determined by their histogenesis, that is, being
either ACA or SCC, which was expected and not sur-
prising. There were only 7 ACA (all but one belonging to
the supervised cluster ACA2, demonstrating expression of
SOX2, markers of stemness and increased proliferation
and apoptosis) found among the SCC-like cluster, all of
which were CK7 positive and showed PAS and alcian-
blue-PASûpositive inclusions in their cytoplasm and nu-
cleus. CK5/6 was expressed by 5/7, and p63 by 4/7, and 2
were positive for TTF1. Interestingly, both TTF1-positive
ACAs were also positive for CK5/6 and p63. For all 7
ACA cases that clustered in the SCC-like cluster, classical
histomorphology combined with mucin positivity, and
CK7 andùif applicableùTTF1 positivity were the de-
termining factors for the diagnosis of ACA. On the op-
posite, 11 SCC (all belonging to the supervised cluster
SCC2) were encountered in the ACA-like cluster. All these
cases were positive for CK5/6 and p63, and negative for
TTF1. These 11 SCC were mostly positive for SOX2 and
showed a pro?le typical of stem cells (CD44+/CD24?).
Obviously, stemness-like signatures of either ACA or SCC

FIGURE 4. Heatmap of hierarchical clustering analysis of the
studied 22 LCC. Patients split into 3 clusters, which are separated by
vertical lines. For markers, light grey indicates lacking (below cut-
off) expression/negativity or CD4/CD8 ratio <1, whereas dark grey
indicates expression/positivity or CD4/CD8 ratio > 1. Only sig-
nificant (P < 0.008) discriminating markers or marker groups as well
as some other informative markers (grade > 1, CD4/CD8 ratio,
CK7, TTF1) are annotated. The grayscale colored heatmap be-
tween the upper dendrogram and the marker heatmap codes for
relevant clinicopathologic data, black indicating patients above
65 years old, male individuals, smokers, patients suffering from
relapses, or higher pathological nodal, pathological tumor, or
pathological Union Internationale Contre le Cancer stages, re-
spectively. Empty spots depict markers that were not analyzable in
respective cases or cases with missing clinicopathologic data. Note
that LCC clearly cluster into 2 groups, which are mainly defined
by expression of either squamous cell/basal markers (?-catenin,
CD44v6, CK5/6, CK34?E12, and/or p63,) or more simple epithelia-
like markers (E-cadherine and/or epithelial cell adhesion molecule
compounds: Ber-EP4, ESA, MOC31) and higher proliferative activ-
ity. CK7 indicates cytokeratin 7; LCC, large cell carcinoma; RHAMM,
receptor for hyaluronan-mediated motility; TTF1, thyroid tran-
scription factor 1.

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

www.appliedimmunohist.com | 7

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Sterlacci et al

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

TABLE 3. Supervised Cluster Analysis According to Histologic Tumor Type

Parameters

Adenocarcinoma (n = 218)

Squamous Cell Carcinoma (n = 125)

Large Cell Carcinoma
(n = 22)

Cluster1
(n = 6)

Grade
Csp3
TGF-?
m/f
mucin
lepidic
p16
p21
EGFR
Ber-EP4
MOC31
ESA

Cluster 2
(n = 192)

Cluster 3
(n = 20)

m/f
Ki67
ALDH1
SOX2
ù
ù
ù
ù
ù
ù
ù
ù

mucin
lepidic
TTF1
EGFR
Ber-EP4
MOC31
ESA
grade
Ki67
CCNE

Cluster 1 (n = 8)
CD24 û /CD44s+
Ber-EP4
MOC31
ESA
MUC5AC
ù
ù
ù
ù
ù
ù
ù

Cluster 2
(n = 45)

Cluster 3
(n = 72)

CD4/CD8
Ber-EP4
MOC31
ESA
CD24 û /CD44s+
OS
GrB+ TIL
ù
ù
ù
ù
ù

GrB+ TIL
CD8+TIL
TIA+ TIL
CBL-B+ TIL
PDL1+ TIM
CD4/CD8
Ber-EP4
MOC31
ESA
CD24 û (CD44s+)
ù
ù

Cluster 1
(n = 9)

CK5/6
CK34?E12
p63
?-catenin
CD44v6
ù
ù
ù
ù
ù
ù
ù

Cluster 2
(n = 13)

Ki67
CCNE
Ber-EP4
MOC31
ESA
RHAMM
E-cadherin
CK5/6
CK34?E12
p63
ù
ù

Bold signi?es increase, normal print signi?es decrease, factors being of signi?cance only when not corrected for multiple testing are in italics.
ALDH1 indicates aldehyde dehydrogenase 1; CBL-B, Casitas B-lineage lymphoma B; CCNE, cyclin E; CD4/CD8, CD4/CD8 ratio; CK, cytokeratin; Csp3, caspase 3;
EGFR, epidermal growth factor; ESA (Ber-EP4, MOC31), epithelial-speci?c antigen; GrB, granzyme B; m/f, male/female ratio; MUC5AC, mucin 5AC; OS, overall
survival; PDL1, programmed death ligand 1; RHAMM, receptor for hyaluronan-mediated motility; SOX2, sex-determining region Y-box 2; TGF-?, transforming growth
factor beta; TIL, tumor-in?ltrating lymphocytes; TIM, tumor-in?ltrating mononuclear cells; TTF1, thyroid transcription factor 1.

cause dif?culties in assigning tumors to a speci?c entity in
clustering analyses.

Interestingly, with all caveats of more limited case
numbers and fair reproducibility, there appears cluster 3.
It encompasses (i) a subgroup of TTF1+ and CK7+ cases,
(ii) another subgroup of p63+ and CK5/6+ tumors, with
or without expression of TTF1 and/or CK7, that primarily
clustered together because of their very high amounts of
intratumoral CD8+ and TGF-?+ (and TIA1+)-positive
lymphocytes, and (iii) a third, small subgroup of CK7+
and mostly TTF1+ cases with expression of pAKT, che-
mokine receptor type 4 (CXCR4), and presence of higher
amounts of TGF-?+ (but not CD8+) lymphocytes that
particularly displayed higher relapse frequencies (because
of further subgrouping, no statistical testing has been
performed for these instances). For this cluster 3, ob-
viously, the signature of the TIL prevailed over the sig-
nature of histogenesis. This may be especially important in
the light of current
immunomodulating therapies for
NSCLC.29 As only a portion of patients respond to such
treatment strategies, a successful therapy may be linked to
the presence of a certain immunologic tumor phenotype/
signature. The important role of intratumoral contents of
CD8+ T cells to de?ne a distinct unsupervised NSCLC
cluster, ?ts with recent data showing that the prognostic
impact of PDL1 expression in NSCLC cases depends on
the presence or absence of CD8+TIL andùtogether with
the very encouraging in vivo data on the ef?cacy of im-
mune checkpoint modulation in subsets of NSCLS
patientsùpoints toward a real biological existence of such
an NSCLC cluster with decelerated but potentially
ôawakenableö immunity.24,30

Importantly, in this unsupervised setting, neither a
neuroendocrine cluster nor an LCC cluster appeared, ?t-
ting with the current concepts of NSCLC classi?cation,
which challenged and to a great part abolished respective

entities, and which is also in line with our previous work as
well as other reports (see also discussion on the supervised
LCC cluster analysis).13,19,27

all

for

Supervised cluster analysis could further identify
clusters within the predetermined NSCLC
relevant
histotypes, that is, ACA, SCC, and LCC (Figs. 2û4 and
Table 3).

(P < 0.000018

For ACA (n = 218), 3 groups were apparent: 1 very
large group, and 2 groups with far less cases (Fig. 2). One
cluster (ACA1, n = 6) was the only 1 with an inversed
male/female ratio with twice as many women than men,
and encompassed cases without mucin and without lepidic
growth, with higher grade and lower expression of p16
and p21, higher active caspase 3 presence, lower to lacking
EGFR and Ep-CAM compounds, and higher tumor
expression of TGF-?
listed
markers). The second smaller cluster (ACA3, n = 20)
contained a higher proportion of cases with detectable
mucin and of cases with lepidic predominance, and, thus,
was of lower grade. Furthermore, this cluster displayed
high TTF1 positivity, a low Ki67 proliferation rate, and
low expression of CCNE, but was linked to the expression
of EGFR and Ep-CAM compounds (P < 0.000018 for all
listed markers). The third, very large ACA cluster (ACA2,
n = 192) showed the highest male predominance, had the
highest Ki67 proliferation rate, and higher ALDH1 and
SOX2 positivity (P < 0.000018 for all
listed markers).
Notably, these results show that a grading system for
ACA according to architecture, particularly lepidic
growth, as previously proposed and reproduced on our
cohort as well, is also re?ected by clustering analysis.19,27
Among the SCC (n = 125) also, 3 main clusters were
identi?ed (Fig. 3). One small group (SCC1, n = 8) was
de?ned by complete lack of the stem cellûlike phenotype
(CD24?/CD44s+), lacking expression of Ep-CAM com-
pounds, and MUC5AC (P < 0.000018 for all
listed

8 | www.appliedimmunohist.com

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

Cluster Analysis According to Immunohistochemistry

clusters, 6 of the 9 LCC1 appeared in the unsupervised cluster
2 (SCC-like) and only 1 and 2 in the unsupervised cluster 1
(ACA-like) and 3 (immune signature), respectively, whereas
11 of the 13 LCC2 emerged in the unsupervised cluster 1
(ACA-like) and 2 in the unsupervised cluster 2 (SCC-like,
P = 0.002). These results suggest that it is generally possible to
speci?cally phenotypically assign LCC
further, more
(diagnosed based on histologic analysis) to favor either
SCC or ACA, which is currently carried out in the routine
diagnosis of NSCLC,
in which TTF1 and p63 or p40
expression is regarded as important as morphology to
classify cases.

In conclusion, we could show that clustering analysis
is generally feasible for NSCLC and is a useful tool for
verifying established characteristics as well as generating
new standpoints. Speci?cally, our data show that tumor
grading of ACA by architecture is signi?cant, that LCC is
presumably not a separate entity, that an immunologic
clusterùdepending on patientsÆ immunity and the com-
plex interaction between the tumor and the lymphocytesù
likely exists, and that one prognostically relevant SCC
cluster with high CD4/CD8 ratio and lower presence of
granzyme B+ TIL and poorer outcomes exists.

REFERENCES
1. Abd El-Rehim DM, Ball G, Pinder SE, et al. High-throughput
protein expression analysis using tissue microarray technology of a
large well-characterised series identifies biologically distinct classes of
breast cancer confirming recent cDNA expression analyses. Int J
Cancer. 2005;116:340û350.

2. Alizadeh AA, Ross DT, Perou CM, et al. Towards a novel
classification of human malignancies based on gene expression
patterns. J Pathol. 2001;195:41û52.

3. G³ler EN. Gene expression profiling in breast cancer and its effect on
therapy selection in early-stage breast cancer. Eur J Breast Healt.
2017;13:168û174.

4. Soria D, Garibaldi JM, Ambrogi F, et al. A methodology to identify
consensus classes from clustering algorithms applied to immunohis-
tochemical data from breast cancer patients. Comput Biol Med.
2010;40:318û330.

5. Tomida S, Koshikawa K, Yatabe Y, et al. Gene expression-based,
individualized outcome prediction for surgically treated lung cancer
patients. Oncogene. 2004;23:5360û5370.

6. Hamamoto J, Soejima K, Yoda S, et al. Identification of micro-
RNAs differentially expressed between lung squamous cell carcino-
ma and lung adenocarcinoma. Mol Med Rep. 2013;8:456û462.
7. Markou A, Sourvinou I, Vorkas PA, et al. Clinical evaluation of
microRNA expression profiling in non small cell lung cancer. Lung
Cancer. 2013;81:388û396.

8. Menter T, Dickenmann M, Juskevicius D, et al. Comprehensive
phenotypic characterization of PTLD reveals potential reliance on
EBV or NF-?B signalling instead of B-cell receptor signalling.
Hematol Oncol. 2017;35:187û197.

9. Menter T, Ernst M, Drachneris J, et al. Phenotype profiling of
primary testicular diffuse large B-cell lymphomas. Hematol Oncol.
2014;32:72û81.

10. Au NH, Cheang M, Huntsman DG, et al. Evaluation of
immunohistochemical markers in non-small cell
lung cancer by
unsupervised hierarchical clustering analysis: a tissue microarray
study of 284 cases and 18 markers. J Pathol. 2004;204:101û109.
11. Grossi F, Spizzo R, Bordo D, et al. Prognostic stratification of stage
IIIA pN2 non-small cell
lung cancer by hierarchical clustering
analysis of tissue microarray immunostaining data: an Alpe Adria
Thoracic Oncology Multidisciplinary Group study (ATOM 014).
J Thorac Oncol. 2010;5:1354û1360.

FIGURE 5. Survival analysis of SCC clusters. Note the poorer out-
come of SCC cluster 2 (linked to high CD4/CD8 ratio, lower pres-
ence of granzyme B+ tumor-infiltrating lymphocytes, higher
expression of epithelial cell adhesion molecule compounds, and
higher expression of stemness markers (ABCG5+, CD24?, D44s+)
compared with cluster 1 and 3 (for simplicity and because of having
almost the same slope, shown together); P = 0.041 (for comparison
between all 3 clusters). ABCG5 indicates ATP binding cassette sub-
family G member 5; SCC, squamous cell carcinoma.

markers). Another cluster (SCC2, n = 45) was linked to
high CD4/CD8 ratio,
lower presence of granzyme B+
TIL, higher expression of Ep-CAM compounds, and the
presence of the proportionally most cases with a stem cellû
like phenotype (CD24?/CD44s+) (P < 0.000018 for all
listed markers), and was the only cluster with a speci?cù
in that case worseùprognosis (Fig. 5). The third cluster
(SCC3, n = 72) showed the opposite with a low CD4/CD8
ratio and a higher presence of granzyme B+ TIL
(P < 0.000018 for all listed markers), andùyet statistical-
ly not signi?cant when corrected for multiple testingù
higher amounts of CD8+ and/or TIA+, CBL-B+ and
PDL1+ TIL, as well as lower expression of the Ep-CAM
compounds and less cases with stem cellûlike phenotype
(P < 0.000018 for the latter two).

For the LCC (n = 22), 2 groups were distinguished by
our cluster analysis (Fig. 4): one with frequent expression of
basal markers (CK5/6, CK34?E12, p63), ?-catenin, and
CD44v6 (LCC1, n = 9), that is, SCC-like, and another with
diminished basal markers but higher Ki67 proliferation rate
and expression of CCNE, as well as increased expression of
Ep-CAM compounds, receptor for hyaluronan-mediated
motility (RHAMM), and E-cadherin (LCC2, n = 13), that
is, more ACA-like. As only 22 LCC entered the analysis, the
marker distribution was assessed in an exploratory setting,
and P-values were corrected for multiple testing of 3
histotypes with 2 clusters and not for 70 markers, that is,
P < 0.008. Importantly, when linked to the unsupervised

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

www.appliedimmunohist.com | 9

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.

Sterlacci et al

Appl Immunohistochem Mol Morphol (cid:1) Volume 00, Number 00, ÆÆ 2019

12. Kocher F, Hilbe W, Seeber A, et al. Longitudinal analysis of 2293
NSCLC patients: a comprehensive study from the TYROL registry.
Lung Cancer. 2015;87:193û200.

13. Sterlacci W, Fiegl M, Hilbe W, et al. Clinical relevance of
neuroendocrine differentiation in non-small cell lung cancer assessed
by immunohistochemistry: a retrospective study on 405 surgically
resected cases. Virchows Arch. 2009;455:125û132.

14. Sterlacci W, Fiegl M, Hilbe W, et al. Deregulation of p27 and cyclin
D1/D3 control over mitosis is associated with unfavorable prognosis
in non-small cell lung cancer, as determined in 405 operated patients.
J Thorac Oncol. 2010;5:1325û1336.

15. Sterlacci W, Tzankov A, Veits L, et al. The prognostic impact of sex
on surgically resected non-small cell
lung cancer depends on
clinicopathologic characteristics. Am J Clin Pathol. 2011;135:
611û618.

16. Sterlacci W, Tzankov A, Veits L, et al. A comprehensive analysis of
p16 expression, gene status, and promoter hypermethylation in
surgically resected non-small cell lung carcinomas. J Thorac Oncol.
2011;6:1649û1657.

17. Sterlacci W, Fiegl M, Tzankov A. Prognostic and predictive value of
cell cycle deregulation in non-small-cell lung cancer. Pathobiology.
2012;79:175û194.

18. Sterlacci W, Wolf D, Savic S, et al. High transforming growth factor
? expression represents an important prognostic parameter for
surgically resected non-small cell lung cancer. Hum Pathol. 2012;43:
339û349.

19. Sterlacci W, Savic S, Schmid T, et al. Tissue-sparing application of
the newly proposed IASLC/ATS/ERS classification of adenocarci-
noma of the lung shows practical diagnostic and prognostic impact.
Am J Clin Pathol. 2012;137:946û956.

20. Sterlacci W, Savic S, Fiegl M, et al. Putative stem cell markers in
lung cancer: a clinicopathologic characterization.

non-small-cell
J Thorac Oncol. 2014;9:41û49.

21. Augustin F, Fiegl M, Schmid T, et al. Receptor for hyaluronic acid-
mediated motility (RHAMM, CD168) expression is prognostically
important in both nodal negative and nodal positive large cell lung
cancer. J Clin Pathol. 2015;68:368û373.

22. Pomme G, Augustin F, Fiegl M, et al. Detailed assessment of
reveals
microvasculature markers
potentially clinically relevant characteristics. Virchows Arch. 2015;467:
55û66.

in non-small cell

lung cancer

23. Sterlacci W, Saker S, Huber B, et al. Expression of the CXCR4
ligand SDF-1/CXCL12 is prognostically important for adenocarci-
noma and large cell carcinoma of the lung. Virchows Arch. 2016;
468:463û471.

24. Sterlacci W, Fiegl M, Droeser RA, et al. Expression of PD-L1
identifies a subgroup of more aggressive non-small cell carcinomas of
the lung. Pathobiology. 2016;83:267û275.

25. Sterlacci W, Fiegl M, Gugger M, et al. MET overexpression and
gene amplification: prevalence, clinico-pathological characteristics
and prognostic significance in a large cohort of patients with
surgically resected NSCLC. Virchows Arch. 2017;471:49û55.

26. Sterlacci W, Fiegl M, Veits L, et al. Diagnostic and prognostic
impact of mucin 1-6 expression in non-small cell lung cancer. Indian
J Pathol Microbiol. 2018;61:187û191.

27. Travis W, Brambilla E, Burke A, et al. WHO Classification of
Tumours of the Lung, Pleura, Thymus and Heart. Lyon: IARCPress;
2015.

28. Sneath PH. Some thoughts on bacterial classification. J Gen Microbiol.

1957;17:184û200.

29. Pabani A, Butts CA. Current landscape of immunotherapy for the
treatment of metastatic non-small-cell lung cancer. Curr Oncol. 2018;
25:S94ûS102.

30. Durgeau A, Virk Y, Corgnac S, et al. Recent advances in targeting
immunity for more effective cancer immunotherapy.

CD8 T-cell
Front Immunol. 2018;9:14.

10 | www.appliedimmunohist.com

Copyright ® 2019 Wolters Kluwer Health, Inc. All rights reserved.

Copyright r 2019 Wolters Kluwer Health, Inc. Unauthorized reproduction of this article is prohibited.


