<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Mendelian Randomization — mendelianrandomization • ClinicoPath</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mendelian Randomization — mendelianrandomization"><meta name="description" content="Mendelian Randomization (MR) uses genetic variants as instrumental
variables to infer causal relationships between exposures and outcomes."><meta property="og:description" content="Mendelian Randomization (MR) uses genetic variants as instrumental
variables to infer causal relationships between exposures and outcomes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ClinicoPath</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.32.72</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../dev/index.html"><span class="fa fa-cogs"></span></a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-test-data" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-database"></span> Test Data</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-test-data"><li><a class="dropdown-item" href="../articles/test-data-catalog.html">Test Data Catalog</a></li>
    <li><a class="dropdown-item" href="../articles/test-data-complete-catalog.html">Complete Catalog (945 files)</a></li>
    <li><a class="dropdown-item" href="../articles/function-reference.html">Function Reference</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="http://twitter.com/serdarbalci"><span class="fa fab fa-twitter fa-lg"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/sbalci/ClinicoPathJamoviModule/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Mendelian Randomization</h1>
      <small class="dont-index">Source: <a href="https://github.com/sbalci/ClinicoPathJamoviModule/blob/HEAD/R/mendelianrandomization.h.R" class="external-link"><code>R/mendelianrandomization.h.R</code></a></small>
      <div class="d-none name"><code>mendelianrandomization.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Mendelian Randomization (MR) uses genetic variants as instrumental
variables to infer causal relationships between exposures and outcomes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mendelianrandomization</span><span class="op">(</span></span>
<span>  <span class="va">exposure_data</span>,</span>
<span>  <span class="va">outcome_data</span>,</span>
<span>  use_same_dataset <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">snp_column</span>,</span>
<span>  <span class="va">beta_exposure</span>,</span>
<span>  <span class="va">se_exposure</span>,</span>
<span>  pval_exposure <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">beta_outcome</span>,</span>
<span>  <span class="va">se_outcome</span>,</span>
<span>  pval_outcome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  eaf_column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pval_threshold <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  clump_r2 <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  clump_kb <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  mr_methods <span class="op">=</span> <span class="st">"main_three"</span>,</span>
<span>  ivw_model <span class="op">=</span> <span class="st">"random"</span>,</span>
<span>  egger_bootstrap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  bootstrap_samples <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  heterogeneity_test <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pleiotropy_test <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  leave_one_out <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  single_snp_analysis <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  mr_presso <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  presso_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  plot_forest <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plot_funnel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plot_scatter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plot_loo <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  harmonize_alleles <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_palindromic <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  steiger_filtering <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min_snps <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  conf_level <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  random_seed <span class="op">=</span> <span class="fl">42</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-exposure-data">exposure_data<a class="anchor" aria-label="anchor" href="#arg-exposure-data"></a></dt>
<dd><p>GWAS summary statistics for exposure variable</p></dd>


<dt id="arg-outcome-data">outcome_data<a class="anchor" aria-label="anchor" href="#arg-outcome-data"></a></dt>
<dd><p>GWAS summary statistics for outcome variable (optional
if same dataset)</p></dd>


<dt id="arg-use-same-dataset">use_same_dataset<a class="anchor" aria-label="anchor" href="#arg-use-same-dataset"></a></dt>
<dd><p>Whether exposure and outcome are in the same
dataset</p></dd>


<dt id="arg-snp-column">snp_column<a class="anchor" aria-label="anchor" href="#arg-snp-column"></a></dt>
<dd><p>Column containing SNP identifiers (e.g., rsID)</p></dd>


<dt id="arg-beta-exposure">beta_exposure<a class="anchor" aria-label="anchor" href="#arg-beta-exposure"></a></dt>
<dd><p>Effect size (beta) for exposure</p></dd>


<dt id="arg-se-exposure">se_exposure<a class="anchor" aria-label="anchor" href="#arg-se-exposure"></a></dt>
<dd><p>Standard error for exposure beta</p></dd>


<dt id="arg-pval-exposure">pval_exposure<a class="anchor" aria-label="anchor" href="#arg-pval-exposure"></a></dt>
<dd><p>P-value for exposure association</p></dd>


<dt id="arg-beta-outcome">beta_outcome<a class="anchor" aria-label="anchor" href="#arg-beta-outcome"></a></dt>
<dd><p>Effect size (beta) for outcome</p></dd>


<dt id="arg-se-outcome">se_outcome<a class="anchor" aria-label="anchor" href="#arg-se-outcome"></a></dt>
<dd><p>Standard error for outcome beta</p></dd>


<dt id="arg-pval-outcome">pval_outcome<a class="anchor" aria-label="anchor" href="#arg-pval-outcome"></a></dt>
<dd><p>P-value for outcome association</p></dd>


<dt id="arg-eaf-column">eaf_column<a class="anchor" aria-label="anchor" href="#arg-eaf-column"></a></dt>
<dd><p>Effect allele frequency (optional)</p></dd>


<dt id="arg-pval-threshold">pval_threshold<a class="anchor" aria-label="anchor" href="#arg-pval-threshold"></a></dt>
<dd><p>Genome-wide significance threshold (default 5e-8)</p></dd>


<dt id="arg-clump-r-">clump_r2<a class="anchor" aria-label="anchor" href="#arg-clump-r-"></a></dt>
<dd><p>R² threshold for LD clumping</p></dd>


<dt id="arg-clump-kb">clump_kb<a class="anchor" aria-label="anchor" href="#arg-clump-kb"></a></dt>
<dd><p>Distance window for LD clumping in kilobases</p></dd>


<dt id="arg-mr-methods">mr_methods<a class="anchor" aria-label="anchor" href="#arg-mr-methods"></a></dt>
<dd><p>Which MR methods to perform</p></dd>


<dt id="arg-ivw-model">ivw_model<a class="anchor" aria-label="anchor" href="#arg-ivw-model"></a></dt>
<dd><p>Model specification for IVW method</p></dd>


<dt id="arg-egger-bootstrap">egger_bootstrap<a class="anchor" aria-label="anchor" href="#arg-egger-bootstrap"></a></dt>
<dd><p>Use bootstrap for MR-Egger confidence intervals</p></dd>


<dt id="arg-bootstrap-samples">bootstrap_samples<a class="anchor" aria-label="anchor" href="#arg-bootstrap-samples"></a></dt>
<dd><p>Number of bootstrap samples</p></dd>


<dt id="arg-heterogeneity-test">heterogeneity_test<a class="anchor" aria-label="anchor" href="#arg-heterogeneity-test"></a></dt>
<dd><p>Perform Cochran Q heterogeneity test</p></dd>


<dt id="arg-pleiotropy-test">pleiotropy_test<a class="anchor" aria-label="anchor" href="#arg-pleiotropy-test"></a></dt>
<dd><p>Test for horizontal pleiotropy (MR-Egger intercept)</p></dd>


<dt id="arg-leave-one-out">leave_one_out<a class="anchor" aria-label="anchor" href="#arg-leave-one-out"></a></dt>
<dd><p>Perform leave-one-out sensitivity analysis</p></dd>


<dt id="arg-single-snp-analysis">single_snp_analysis<a class="anchor" aria-label="anchor" href="#arg-single-snp-analysis"></a></dt>
<dd><p>Analyze each SNP individually</p></dd>


<dt id="arg-mr-presso">mr_presso<a class="anchor" aria-label="anchor" href="#arg-mr-presso"></a></dt>
<dd><p>Detect and correct for outliers using MR-PRESSO</p></dd>


<dt id="arg-presso-threshold">presso_threshold<a class="anchor" aria-label="anchor" href="#arg-presso-threshold"></a></dt>
<dd><p>P-value threshold for MR-PRESSO outlier detection</p></dd>


<dt id="arg-plot-forest">plot_forest<a class="anchor" aria-label="anchor" href="#arg-plot-forest"></a></dt>
<dd><p>Generate forest plot of MR estimates</p></dd>


<dt id="arg-plot-funnel">plot_funnel<a class="anchor" aria-label="anchor" href="#arg-plot-funnel"></a></dt>
<dd><p>Generate funnel plot for asymmetry assessment</p></dd>


<dt id="arg-plot-scatter">plot_scatter<a class="anchor" aria-label="anchor" href="#arg-plot-scatter"></a></dt>
<dd><p>Generate scatter plot with method lines</p></dd>


<dt id="arg-plot-loo">plot_loo<a class="anchor" aria-label="anchor" href="#arg-plot-loo"></a></dt>
<dd><p>Plot leave-one-out analysis results</p></dd>


<dt id="arg-harmonize-alleles">harmonize_alleles<a class="anchor" aria-label="anchor" href="#arg-harmonize-alleles"></a></dt>
<dd><p>Harmonize exposure and outcome effect alleles</p></dd>


<dt id="arg-remove-palindromic">remove_palindromic<a class="anchor" aria-label="anchor" href="#arg-remove-palindromic"></a></dt>
<dd><p>Remove palindromic SNPs with ambiguous strand</p></dd>


<dt id="arg-steiger-filtering">steiger_filtering<a class="anchor" aria-label="anchor" href="#arg-steiger-filtering"></a></dt>
<dd><p>Filter SNPs using Steiger test for directionality</p></dd>


<dt id="arg-min-snps">min_snps<a class="anchor" aria-label="anchor" href="#arg-min-snps"></a></dt>
<dd><p>Minimum number of SNPs required for analysis</p></dd>


<dt id="arg-conf-level">conf_level<a class="anchor" aria-label="anchor" href="#arg-conf-level"></a></dt>
<dd><p>Confidence level for intervals</p></dd>


<dt id="arg-random-seed">random_seed<a class="anchor" aria-label="anchor" href="#arg-random-seed"></a></dt>
<dd><p>Random seed for reproducibility</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A results object containing:</p><table class="table table"><tr><td><code>results$instructions</code></td><td></td><td></td><td></td><td></td><td>a html</td></tr><tr><td><code>results$dataInfo</code></td><td></td><td></td><td></td><td></td><td>a preformatted</td></tr><tr><td><code>results$snpSelection</code></td><td></td><td></td><td></td><td></td><td>a html</td></tr><tr><td><code>results$selectedSNPs</code></td><td></td><td></td><td></td><td></td><td>a table</td></tr><tr><td><code>results$mrResults</code></td><td></td><td></td><td></td><td></td><td>a table</td></tr><tr><td><code>results$heterogeneityTable</code></td><td></td><td></td><td></td><td></td><td>a table</td></tr><tr><td><code>results$pleiotropyTable</code></td><td></td><td></td><td></td><td></td><td>a table</td></tr><tr><td><code>results$looTable</code></td><td></td><td></td><td></td><td></td><td>a table</td></tr><tr><td><code>results$pressoResults</code></td><td></td><td></td><td></td><td></td><td>a html</td></tr><tr><td><code>results$forestPlot</code></td><td></td><td></td><td></td><td></td><td>an image</td></tr><tr><td><code>results$funnelPlot</code></td><td></td><td></td><td></td><td></td><td>an image</td></tr><tr><td><code>results$scatterPlot</code></td><td></td><td></td><td></td><td></td><td>an image</td></tr><tr><td><code>results$looPlot</code></td><td></td><td></td><td></td><td></td><td>an image</td></tr><tr><td><code>results$interpretation</code></td><td></td><td></td><td></td><td></td><td>a html</td></tr><tr><td><code>results$assumptions</code></td><td></td><td></td><td></td><td></td><td>a html</td></tr></table><p>Tables can be converted to data frames with <code>asDF</code> or <code><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></code>. For example:</p>
<p><code>results$selectedSNPs$asDF</code></p>
<p><code>as.data.frame(results$selectedSNPs)</code></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This module supports GWAS summary statistics (NOT individual-level data)
and provides multiple MR methods with comprehensive sensitivity analysis.</p>
<p>Applications: Genetic epidemiology, drug target validation, causal
inference in observational studies.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example: Analyze effect of BMI on coronary disease</span></span></span>
<span class="r-in"><span><span class="fu">mendelianrandomization</span><span class="op">(</span></span></span>
<span class="r-in"><span>    exposure_data <span class="op">=</span> <span class="va">bmi_gwas</span>,</span></span>
<span class="r-in"><span>    outcome_data <span class="op">=</span> <span class="va">cad_gwas</span>,</span></span>
<span class="r-in"><span>    snp_column <span class="op">=</span> <span class="st">'rsID'</span>,</span></span>
<span class="r-in"><span>    beta_exposure <span class="op">=</span> <span class="st">'beta_bmi'</span>,</span></span>
<span class="r-in"><span>    se_exposure <span class="op">=</span> <span class="st">'se_bmi'</span>,</span></span>
<span class="r-in"><span>    beta_outcome <span class="op">=</span> <span class="st">'beta_cad'</span>,</span></span>
<span class="r-in"><span>    se_outcome <span class="op">=</span> <span class="st">'se_cad'</span>,</span></span>
<span class="r-in"><span>    mr_methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'IVW'</span>, <span class="st">'MR-Egger'</span>, <span class="st">'Weighted Median'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in mendelianrandomization(exposure_data = bmi_gwas, outcome_data = cad_gwas,     snp_column = "rsID", beta_exposure = "beta_bmi", se_exposure = "se_bmi",     beta_outcome = "beta_cad", se_outcome = "se_cad", mr_methods = c("IVW",         "MR-Egger", "Weighted Median")):</span> 'missing(data)' did not find an argument</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.serdarbalci.com/" class="external-link">Serdar Balci</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

