
# This file is automatically generated, you probably don't want to edit this

biomarkerdiscoveryOptions <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "biomarkerdiscoveryOptions",
    inherit = jmvcore::Options,
    public = list(
        initialize = function(
            outcome_var = NULL,
            biomarker_vars = NULL,
            clinical_vars = NULL,
            batch_var = NULL,
            patient_id = NULL,
            time_var = NULL,
            event_var = NULL,
            discovery_method = "elastic_net",
            outcome_type = "binary",
            data_type = "genomics",
            data_preprocessing = TRUE,
            normalization_method = "z_score",
            batch_correction = FALSE,
            batch_method = "combat",
            filter_low_variance = TRUE,
            variance_threshold = 0.1,
            feature_selection_method = "univariate_stats",
            n_features_select = 50,
            fdr_threshold = 0.05,
            correlation_threshold = 0.9,
            validation_method = "cv_10fold",
            train_proportion = 0.7,
            hyperparameter_tuning = TRUE,
            n_bootstrap_samples = 1000,
            random_seed = 42,
            biomarker_ranking = TRUE,
            stability_analysis = TRUE,
            clinical_performance = TRUE,
            signature_development = TRUE,
            pathway_analysis = FALSE,
            interpretability = TRUE,
            shap_analysis = TRUE,
            lime_analysis = FALSE,
            feature_interaction = TRUE,
            partial_dependence = TRUE,
            biomarker_networks = FALSE,
            cutpoint_optimization = TRUE,
            risk_stratification = TRUE,
            nomogram_development = TRUE,
            decision_curve_analysis = TRUE,
            external_validation = TRUE,
            biomarker_generalizability = TRUE,
            robustness_testing = TRUE,
            quality_control = TRUE,
            outlier_detection = TRUE,
            missing_data_analysis = TRUE,
            detailed_results = TRUE,
            biomarker_report = TRUE,
            export_biomarkers = FALSE,
            save_signature = FALSE,
            regulatory_documentation = TRUE, ...) {

            super$initialize(
                package="ClinicoPath",
                name="biomarkerdiscovery",
                requiresData=TRUE,
                ...)

            private$..outcome_var <- jmvcore::OptionVariable$new(
                "outcome_var",
                outcome_var,
                suggested=list(
                    "nominal",
                    "continuous"),
                permitted=list(
                    "numeric",
                    "factor"))
            private$..biomarker_vars <- jmvcore::OptionVariables$new(
                "biomarker_vars",
                biomarker_vars,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..clinical_vars <- jmvcore::OptionVariables$new(
                "clinical_vars",
                clinical_vars,
                suggested=list(
                    "continuous",
                    "nominal"),
                permitted=list(
                    "numeric",
                    "factor"))
            private$..batch_var <- jmvcore::OptionVariable$new(
                "batch_var",
                batch_var,
                suggested=list(
                    "nominal"),
                permitted=list(
                    "factor"))
            private$..patient_id <- jmvcore::OptionVariable$new(
                "patient_id",
                patient_id,
                suggested=list(
                    "nominal"),
                permitted=list(
                    "factor",
                    "id"))
            private$..time_var <- jmvcore::OptionVariable$new(
                "time_var",
                time_var,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..event_var <- jmvcore::OptionVariable$new(
                "event_var",
                event_var,
                suggested=list(
                    "nominal"),
                permitted=list(
                    "factor"))
            private$..discovery_method <- jmvcore::OptionList$new(
                "discovery_method",
                discovery_method,
                options=list(
                    "elastic_net",
                    "random_forest",
                    "gradient_boosting",
                    "svm_rfe",
                    "univariate",
                    "multivariate",
                    "ensemble"),
                default="elastic_net")
            private$..outcome_type <- jmvcore::OptionList$new(
                "outcome_type",
                outcome_type,
                options=list(
                    "binary",
                    "continuous",
                    "survival",
                    "multiclass"),
                default="binary")
            private$..data_type <- jmvcore::OptionList$new(
                "data_type",
                data_type,
                options=list(
                    "genomics",
                    "proteomics",
                    "metabolomics",
                    "clinical",
                    "mixed_omics"),
                default="genomics")
            private$..data_preprocessing <- jmvcore::OptionBool$new(
                "data_preprocessing",
                data_preprocessing,
                default=TRUE)
            private$..normalization_method <- jmvcore::OptionList$new(
                "normalization_method",
                normalization_method,
                options=list(
                    "z_score",
                    "min_max",
                    "robust",
                    "quantile",
                    "log_transform"),
                default="z_score")
            private$..batch_correction <- jmvcore::OptionBool$new(
                "batch_correction",
                batch_correction,
                default=FALSE)
            private$..batch_method <- jmvcore::OptionList$new(
                "batch_method",
                batch_method,
                options=list(
                    "combat",
                    "limma",
                    "pca"),
                default="combat")
            private$..filter_low_variance <- jmvcore::OptionBool$new(
                "filter_low_variance",
                filter_low_variance,
                default=TRUE)
            private$..variance_threshold <- jmvcore::OptionNumber$new(
                "variance_threshold",
                variance_threshold,
                min=0,
                max=1,
                default=0.1)
            private$..feature_selection_method <- jmvcore::OptionList$new(
                "feature_selection_method",
                feature_selection_method,
                options=list(
                    "univariate_stats",
                    "correlation_filter",
                    "mutual_information",
                    "recursive_elimination",
                    "stability_selection",
                    "boruta"),
                default="univariate_stats")
            private$..n_features_select <- jmvcore::OptionInteger$new(
                "n_features_select",
                n_features_select,
                min=1,
                max=1000,
                default=50)
            private$..fdr_threshold <- jmvcore::OptionNumber$new(
                "fdr_threshold",
                fdr_threshold,
                min=0.001,
                max=0.5,
                default=0.05)
            private$..correlation_threshold <- jmvcore::OptionNumber$new(
                "correlation_threshold",
                correlation_threshold,
                min=0.5,
                max=0.99,
                default=0.9)
            private$..validation_method <- jmvcore::OptionList$new(
                "validation_method",
                validation_method,
                options=list(
                    "cv_10fold",
                    "cv_5fold",
                    "bootstrap",
                    "holdout",
                    "nested_cv"),
                default="cv_10fold")
            private$..train_proportion <- jmvcore::OptionNumber$new(
                "train_proportion",
                train_proportion,
                min=0.5,
                max=0.9,
                default=0.7)
            private$..hyperparameter_tuning <- jmvcore::OptionBool$new(
                "hyperparameter_tuning",
                hyperparameter_tuning,
                default=TRUE)
            private$..n_bootstrap_samples <- jmvcore::OptionInteger$new(
                "n_bootstrap_samples",
                n_bootstrap_samples,
                min=100,
                max=2000,
                default=1000)
            private$..random_seed <- jmvcore::OptionInteger$new(
                "random_seed",
                random_seed,
                min=1,
                max=999999,
                default=42)
            private$..biomarker_ranking <- jmvcore::OptionBool$new(
                "biomarker_ranking",
                biomarker_ranking,
                default=TRUE)
            private$..stability_analysis <- jmvcore::OptionBool$new(
                "stability_analysis",
                stability_analysis,
                default=TRUE)
            private$..clinical_performance <- jmvcore::OptionBool$new(
                "clinical_performance",
                clinical_performance,
                default=TRUE)
            private$..signature_development <- jmvcore::OptionBool$new(
                "signature_development",
                signature_development,
                default=TRUE)
            private$..pathway_analysis <- jmvcore::OptionBool$new(
                "pathway_analysis",
                pathway_analysis,
                default=FALSE)
            private$..interpretability <- jmvcore::OptionBool$new(
                "interpretability",
                interpretability,
                default=TRUE)
            private$..shap_analysis <- jmvcore::OptionBool$new(
                "shap_analysis",
                shap_analysis,
                default=TRUE)
            private$..lime_analysis <- jmvcore::OptionBool$new(
                "lime_analysis",
                lime_analysis,
                default=FALSE)
            private$..feature_interaction <- jmvcore::OptionBool$new(
                "feature_interaction",
                feature_interaction,
                default=TRUE)
            private$..partial_dependence <- jmvcore::OptionBool$new(
                "partial_dependence",
                partial_dependence,
                default=TRUE)
            private$..biomarker_networks <- jmvcore::OptionBool$new(
                "biomarker_networks",
                biomarker_networks,
                default=FALSE)
            private$..cutpoint_optimization <- jmvcore::OptionBool$new(
                "cutpoint_optimization",
                cutpoint_optimization,
                default=TRUE)
            private$..risk_stratification <- jmvcore::OptionBool$new(
                "risk_stratification",
                risk_stratification,
                default=TRUE)
            private$..nomogram_development <- jmvcore::OptionBool$new(
                "nomogram_development",
                nomogram_development,
                default=TRUE)
            private$..decision_curve_analysis <- jmvcore::OptionBool$new(
                "decision_curve_analysis",
                decision_curve_analysis,
                default=TRUE)
            private$..external_validation <- jmvcore::OptionBool$new(
                "external_validation",
                external_validation,
                default=TRUE)
            private$..biomarker_generalizability <- jmvcore::OptionBool$new(
                "biomarker_generalizability",
                biomarker_generalizability,
                default=TRUE)
            private$..robustness_testing <- jmvcore::OptionBool$new(
                "robustness_testing",
                robustness_testing,
                default=TRUE)
            private$..quality_control <- jmvcore::OptionBool$new(
                "quality_control",
                quality_control,
                default=TRUE)
            private$..outlier_detection <- jmvcore::OptionBool$new(
                "outlier_detection",
                outlier_detection,
                default=TRUE)
            private$..missing_data_analysis <- jmvcore::OptionBool$new(
                "missing_data_analysis",
                missing_data_analysis,
                default=TRUE)
            private$..detailed_results <- jmvcore::OptionBool$new(
                "detailed_results",
                detailed_results,
                default=TRUE)
            private$..biomarker_report <- jmvcore::OptionBool$new(
                "biomarker_report",
                biomarker_report,
                default=TRUE)
            private$..export_biomarkers <- jmvcore::OptionBool$new(
                "export_biomarkers",
                export_biomarkers,
                default=FALSE)
            private$..save_signature <- jmvcore::OptionBool$new(
                "save_signature",
                save_signature,
                default=FALSE)
            private$..regulatory_documentation <- jmvcore::OptionBool$new(
                "regulatory_documentation",
                regulatory_documentation,
                default=TRUE)

            self$.addOption(private$..outcome_var)
            self$.addOption(private$..biomarker_vars)
            self$.addOption(private$..clinical_vars)
            self$.addOption(private$..batch_var)
            self$.addOption(private$..patient_id)
            self$.addOption(private$..time_var)
            self$.addOption(private$..event_var)
            self$.addOption(private$..discovery_method)
            self$.addOption(private$..outcome_type)
            self$.addOption(private$..data_type)
            self$.addOption(private$..data_preprocessing)
            self$.addOption(private$..normalization_method)
            self$.addOption(private$..batch_correction)
            self$.addOption(private$..batch_method)
            self$.addOption(private$..filter_low_variance)
            self$.addOption(private$..variance_threshold)
            self$.addOption(private$..feature_selection_method)
            self$.addOption(private$..n_features_select)
            self$.addOption(private$..fdr_threshold)
            self$.addOption(private$..correlation_threshold)
            self$.addOption(private$..validation_method)
            self$.addOption(private$..train_proportion)
            self$.addOption(private$..hyperparameter_tuning)
            self$.addOption(private$..n_bootstrap_samples)
            self$.addOption(private$..random_seed)
            self$.addOption(private$..biomarker_ranking)
            self$.addOption(private$..stability_analysis)
            self$.addOption(private$..clinical_performance)
            self$.addOption(private$..signature_development)
            self$.addOption(private$..pathway_analysis)
            self$.addOption(private$..interpretability)
            self$.addOption(private$..shap_analysis)
            self$.addOption(private$..lime_analysis)
            self$.addOption(private$..feature_interaction)
            self$.addOption(private$..partial_dependence)
            self$.addOption(private$..biomarker_networks)
            self$.addOption(private$..cutpoint_optimization)
            self$.addOption(private$..risk_stratification)
            self$.addOption(private$..nomogram_development)
            self$.addOption(private$..decision_curve_analysis)
            self$.addOption(private$..external_validation)
            self$.addOption(private$..biomarker_generalizability)
            self$.addOption(private$..robustness_testing)
            self$.addOption(private$..quality_control)
            self$.addOption(private$..outlier_detection)
            self$.addOption(private$..missing_data_analysis)
            self$.addOption(private$..detailed_results)
            self$.addOption(private$..biomarker_report)
            self$.addOption(private$..export_biomarkers)
            self$.addOption(private$..save_signature)
            self$.addOption(private$..regulatory_documentation)
        }),
    active = list(
        outcome_var = function() private$..outcome_var$value,
        biomarker_vars = function() private$..biomarker_vars$value,
        clinical_vars = function() private$..clinical_vars$value,
        batch_var = function() private$..batch_var$value,
        patient_id = function() private$..patient_id$value,
        time_var = function() private$..time_var$value,
        event_var = function() private$..event_var$value,
        discovery_method = function() private$..discovery_method$value,
        outcome_type = function() private$..outcome_type$value,
        data_type = function() private$..data_type$value,
        data_preprocessing = function() private$..data_preprocessing$value,
        normalization_method = function() private$..normalization_method$value,
        batch_correction = function() private$..batch_correction$value,
        batch_method = function() private$..batch_method$value,
        filter_low_variance = function() private$..filter_low_variance$value,
        variance_threshold = function() private$..variance_threshold$value,
        feature_selection_method = function() private$..feature_selection_method$value,
        n_features_select = function() private$..n_features_select$value,
        fdr_threshold = function() private$..fdr_threshold$value,
        correlation_threshold = function() private$..correlation_threshold$value,
        validation_method = function() private$..validation_method$value,
        train_proportion = function() private$..train_proportion$value,
        hyperparameter_tuning = function() private$..hyperparameter_tuning$value,
        n_bootstrap_samples = function() private$..n_bootstrap_samples$value,
        random_seed = function() private$..random_seed$value,
        biomarker_ranking = function() private$..biomarker_ranking$value,
        stability_analysis = function() private$..stability_analysis$value,
        clinical_performance = function() private$..clinical_performance$value,
        signature_development = function() private$..signature_development$value,
        pathway_analysis = function() private$..pathway_analysis$value,
        interpretability = function() private$..interpretability$value,
        shap_analysis = function() private$..shap_analysis$value,
        lime_analysis = function() private$..lime_analysis$value,
        feature_interaction = function() private$..feature_interaction$value,
        partial_dependence = function() private$..partial_dependence$value,
        biomarker_networks = function() private$..biomarker_networks$value,
        cutpoint_optimization = function() private$..cutpoint_optimization$value,
        risk_stratification = function() private$..risk_stratification$value,
        nomogram_development = function() private$..nomogram_development$value,
        decision_curve_analysis = function() private$..decision_curve_analysis$value,
        external_validation = function() private$..external_validation$value,
        biomarker_generalizability = function() private$..biomarker_generalizability$value,
        robustness_testing = function() private$..robustness_testing$value,
        quality_control = function() private$..quality_control$value,
        outlier_detection = function() private$..outlier_detection$value,
        missing_data_analysis = function() private$..missing_data_analysis$value,
        detailed_results = function() private$..detailed_results$value,
        biomarker_report = function() private$..biomarker_report$value,
        export_biomarkers = function() private$..export_biomarkers$value,
        save_signature = function() private$..save_signature$value,
        regulatory_documentation = function() private$..regulatory_documentation$value),
    private = list(
        ..outcome_var = NA,
        ..biomarker_vars = NA,
        ..clinical_vars = NA,
        ..batch_var = NA,
        ..patient_id = NA,
        ..time_var = NA,
        ..event_var = NA,
        ..discovery_method = NA,
        ..outcome_type = NA,
        ..data_type = NA,
        ..data_preprocessing = NA,
        ..normalization_method = NA,
        ..batch_correction = NA,
        ..batch_method = NA,
        ..filter_low_variance = NA,
        ..variance_threshold = NA,
        ..feature_selection_method = NA,
        ..n_features_select = NA,
        ..fdr_threshold = NA,
        ..correlation_threshold = NA,
        ..validation_method = NA,
        ..train_proportion = NA,
        ..hyperparameter_tuning = NA,
        ..n_bootstrap_samples = NA,
        ..random_seed = NA,
        ..biomarker_ranking = NA,
        ..stability_analysis = NA,
        ..clinical_performance = NA,
        ..signature_development = NA,
        ..pathway_analysis = NA,
        ..interpretability = NA,
        ..shap_analysis = NA,
        ..lime_analysis = NA,
        ..feature_interaction = NA,
        ..partial_dependence = NA,
        ..biomarker_networks = NA,
        ..cutpoint_optimization = NA,
        ..risk_stratification = NA,
        ..nomogram_development = NA,
        ..decision_curve_analysis = NA,
        ..external_validation = NA,
        ..biomarker_generalizability = NA,
        ..robustness_testing = NA,
        ..quality_control = NA,
        ..outlier_detection = NA,
        ..missing_data_analysis = NA,
        ..detailed_results = NA,
        ..biomarker_report = NA,
        ..export_biomarkers = NA,
        ..save_signature = NA,
        ..regulatory_documentation = NA)
)

biomarkerdiscoveryResults <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "biomarkerdiscoveryResults",
    inherit = jmvcore::Group,
    active = list(
        discovery_overview = function() private$.items[["discovery_overview"]],
        data_summary = function() private$.items[["data_summary"]],
        quality_control_summary = function() private$.items[["quality_control_summary"]],
        outlier_analysis = function() private$.items[["outlier_analysis"]],
        feature_selection_summary = function() private$.items[["feature_selection_summary"]],
        selected_biomarkers = function() private$.items[["selected_biomarkers"]],
        discovery_performance = function() private$.items[["discovery_performance"]],
        signature_performance = function() private$.items[["signature_performance"]],
        biomarker_ranking = function() private$.items[["biomarker_ranking"]],
        stability_analysis_results = function() private$.items[["stability_analysis_results"]],
        shap_biomarker_importance = function() private$.items[["shap_biomarker_importance"]],
        biomarker_interactions = function() private$.items[["biomarker_interactions"]],
        optimal_cutpoints = function() private$.items[["optimal_cutpoints"]],
        risk_stratification_results = function() private$.items[["risk_stratification_results"]],
        decision_curve_results = function() private$.items[["decision_curve_results"]],
        pathway_enrichment = function() private$.items[["pathway_enrichment"]],
        cross_validation_results = function() private$.items[["cross_validation_results"]],
        generalizability_assessment = function() private$.items[["generalizability_assessment"]],
        clinical_interpretation = function() private$.items[["clinical_interpretation"]],
        regulatory_summary = function() private$.items[["regulatory_summary"]],
        biomarker_importance_plot = function() private$.items[["biomarker_importance_plot"]],
        roc_comparison_plot = function() private$.items[["roc_comparison_plot"]],
        shap_summary_plot = function() private$.items[["shap_summary_plot"]],
        shap_dependence_plot = function() private$.items[["shap_dependence_plot"]],
        stability_plot = function() private$.items[["stability_plot"]],
        biomarker_correlation_plot = function() private$.items[["biomarker_correlation_plot"]],
        risk_stratification_plot = function() private$.items[["risk_stratification_plot"]],
        decision_curve_plot = function() private$.items[["decision_curve_plot"]],
        pathway_network_plot = function() private$.items[["pathway_network_plot"]],
        biomarker_distribution_plot = function() private$.items[["biomarker_distribution_plot"]],
        nomogram_plot = function() private$.items[["nomogram_plot"]]),
    private = list(),
    public=list(
        initialize=function(options) {
            super$initialize(
                options=options,
                name="",
                title="Biomarker Discovery Platform with ML Interpretability",
                refs=list(
                    "ClinicoPathJamoviModule"))
            self$add(jmvcore::Table$new(
                options=options,
                name="discovery_overview",
                title="Biomarker Discovery Overview",
                columns=list(
                    list(
                        `name`="characteristic", 
                        `title`="Characteristic", 
                        `type`="text"),
                    list(
                        `name`="value", 
                        `title`="Value", 
                        `type`="text")),
                clearWith=list(
                    "outcome_var",
                    "biomarker_vars",
                    "discovery_method")))
            self$add(jmvcore::Table$new(
                options=options,
                name="data_summary",
                title="Dataset Summary",
                columns=list(
                    list(
                        `name`="data_aspect", 
                        `title`="Data Aspect", 
                        `type`="text"),
                    list(
                        `name`="original", 
                        `title`="Original", 
                        `type`="text"),
                    list(
                        `name`="after_preprocessing", 
                        `title`="After Preprocessing", 
                        `type`="text"),
                    list(
                        `name`="final_analysis", 
                        `title`="Final Analysis", 
                        `type`="text")),
                clearWith=list(
                    "biomarker_vars",
                    "data_preprocessing")))
            self$add(jmvcore::Table$new(
                options=options,
                name="quality_control_summary",
                title="Quality Control Summary",
                visible="(quality_control)",
                columns=list(
                    list(
                        `name`="qc_metric", 
                        `title`="QC Metric", 
                        `type`="text"),
                    list(
                        `name`="threshold", 
                        `title`="Threshold", 
                        `type`="text"),
                    list(
                        `name`="passed", 
                        `title`="Passed", 
                        `type`="integer"),
                    list(
                        `name`="failed", 
                        `title`="Failed", 
                        `type`="integer"),
                    list(
                        `name`="action_taken", 
                        `title`="Action Taken", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="outlier_analysis",
                title="Outlier Analysis Results",
                visible="(outlier_detection)",
                columns=list(
                    list(
                        `name`="sample_id", 
                        `title`="Sample ID", 
                        `type`="text"),
                    list(
                        `name`="outlier_score", 
                        `title`="Outlier Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="outlier_type", 
                        `title`="Outlier Type", 
                        `type`="text"),
                    list(
                        `name`="affected_biomarkers", 
                        `title`="Affected Biomarkers", 
                        `type`="integer"),
                    list(
                        `name`="recommendation", 
                        `title`="Recommendation", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="feature_selection_summary",
                title="Feature Selection Summary",
                columns=list(
                    list(
                        `name`="selection_step", 
                        `title`="Selection Step", 
                        `type`="text"),
                    list(
                        `name`="features_input", 
                        `title`="Features Input", 
                        `type`="integer"),
                    list(
                        `name`="features_output", 
                        `title`="Features Selected", 
                        `type`="integer"),
                    list(
                        `name`="selection_criteria", 
                        `title`="Selection Criteria", 
                        `type`="text"),
                    list(
                        `name`="threshold_used", 
                        `title`="Threshold Used", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="selected_biomarkers",
                title="Selected Biomarkers",
                columns=list(
                    list(
                        `name`="biomarker", 
                        `title`="Biomarker", 
                        `type`="text"),
                    list(
                        `name`="importance_score", 
                        `title`="Importance Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="fdr_adjusted_p", 
                        `title`="FDR Adjusted P", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="effect_size", 
                        `title`="Effect Size", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="stability_index", 
                        `title`="Stability Index", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="clinical_relevance", 
                        `title`="Clinical Relevance", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="discovery_performance",
                title="Biomarker Discovery Performance",
                columns=list(
                    list(
                        `name`="metric", 
                        `title`="Metric", 
                        `type`="text"),
                    list(
                        `name`="training", 
                        `title`="Training", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="validation", 
                        `title`="Cross-Validation", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="bootstrap_mean", 
                        `title`="Bootstrap Mean", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="ci_lower", 
                        `title`="95% CI Lower", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="ci_upper", 
                        `title`="95% CI Upper", 
                        `type`="number", 
                        `format`="zto"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="signature_performance",
                title="Biomarker Signature Performance",
                visible="(signature_development)",
                columns=list(
                    list(
                        `name`="signature_name", 
                        `title`="Signature", 
                        `type`="text"),
                    list(
                        `name`="n_biomarkers", 
                        `title`="N Biomarkers", 
                        `type`="integer"),
                    list(
                        `name`="auc_roc", 
                        `title`="AUC-ROC", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="sensitivity", 
                        `title`="Sensitivity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="specificity", 
                        `title`="Specificity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="accuracy", 
                        `title`="Accuracy", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="clinical_impact", 
                        `title`="Clinical Impact Score", 
                        `type`="number", 
                        `format`="zto"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="biomarker_ranking",
                title="Comprehensive Biomarker Ranking",
                visible="(biomarker_ranking)",
                columns=list(
                    list(
                        `name`="rank", 
                        `title`="Rank", 
                        `type`="integer"),
                    list(
                        `name`="biomarker", 
                        `title`="Biomarker", 
                        `type`="text"),
                    list(
                        `name`="composite_score", 
                        `title`="Composite Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="statistical_significance", 
                        `title`="Statistical Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="effect_size_score", 
                        `title`="Effect Size Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="stability_score", 
                        `title`="Stability Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="clinical_utility_score", 
                        `title`="Clinical Utility Score", 
                        `type`="number", 
                        `format`="zto"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="stability_analysis_results",
                title="Biomarker Selection Stability",
                visible="(stability_analysis)",
                columns=list(
                    list(
                        `name`="biomarker", 
                        `title`="Biomarker", 
                        `type`="text"),
                    list(
                        `name`="selection_frequency", 
                        `title`="Selection Frequency", 
                        `type`="number", 
                        `format`="pc"),
                    list(
                        `name`="rank_stability", 
                        `title`="Rank Stability", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="coefficient_stability", 
                        `title`="Coefficient Stability", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="stability_classification", 
                        `title`="Stability Class", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="shap_biomarker_importance",
                title="SHAP Biomarker Importance",
                visible="(shap_analysis)",
                columns=list(
                    list(
                        `name`="biomarker", 
                        `title`="Biomarker", 
                        `type`="text"),
                    list(
                        `name`="mean_shap_value", 
                        `title`="Mean |SHAP|", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="shap_std", 
                        `title`="SHAP Std Dev", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="positive_impact_freq", 
                        `title`="Positive Impact %", 
                        `type`="number", 
                        `format`="pc"),
                    list(
                        `name`="negative_impact_freq", 
                        `title`="Negative Impact %", 
                        `type`="number", 
                        `format`="pc"),
                    list(
                        `name`="shap_rank", 
                        `title`="SHAP Rank", 
                        `type`="integer"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="biomarker_interactions",
                title="Biomarker Interaction Analysis",
                visible="(feature_interaction)",
                columns=list(
                    list(
                        `name`="biomarker_1", 
                        `title`="Biomarker 1", 
                        `type`="text"),
                    list(
                        `name`="biomarker_2", 
                        `title`="Biomarker 2", 
                        `type`="text"),
                    list(
                        `name`="interaction_strength", 
                        `title`="Interaction Strength", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="interaction_type", 
                        `title`="Interaction Type", 
                        `type`="text"),
                    list(
                        `name`="clinical_relevance", 
                        `title`="Clinical Relevance", 
                        `type`="text"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="optimal_cutpoints",
                title="Optimal Cutpoint Analysis",
                visible="(cutpoint_optimization)",
                columns=list(
                    list(
                        `name`="biomarker", 
                        `title`="Biomarker", 
                        `type`="text"),
                    list(
                        `name`="optimal_cutpoint", 
                        `title`="Optimal Cutpoint", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="method", 
                        `title`="Optimization Method", 
                        `type`="text"),
                    list(
                        `name`="sensitivity", 
                        `title`="Sensitivity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="specificity", 
                        `title`="Specificity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="youden_index", 
                        `title`="Youden Index", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="clinical_interpretation", 
                        `title`="Clinical Interpretation", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="risk_stratification_results",
                title="Risk Stratification Analysis",
                visible="(risk_stratification)",
                columns=list(
                    list(
                        `name`="risk_group", 
                        `title`="Risk Group", 
                        `type`="text"),
                    list(
                        `name`="biomarker_score_range", 
                        `title`="Biomarker Score Range", 
                        `type`="text"),
                    list(
                        `name`="n_patients", 
                        `title`="N Patients", 
                        `type`="integer"),
                    list(
                        `name`="event_rate", 
                        `title`="Event Rate", 
                        `type`="number", 
                        `format`="pc"),
                    list(
                        `name`="hazard_ratio", 
                        `title`="Hazard Ratio", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="ci_lower", 
                        `title`="HR 95% CI Lower", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="ci_upper", 
                        `title`="HR 95% CI Upper", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="decision_curve_results",
                title="Decision Curve Analysis Results",
                visible="(decision_curve_analysis)",
                columns=list(
                    list(
                        `name`="threshold_probability", 
                        `title`="Threshold Probability", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="net_benefit_biomarkers", 
                        `title`="Net Benefit (Biomarkers)", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="net_benefit_treat_all", 
                        `title`="Net Benefit (Treat All)", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="net_benefit_treat_none", 
                        `title`="Net Benefit (Treat None)", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="clinical_impact", 
                        `title`="Clinical Impact", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="pathway_enrichment",
                title="Pathway Enrichment Analysis",
                visible="(pathway_analysis)",
                columns=list(
                    list(
                        `name`="pathway_name", 
                        `title`="Pathway", 
                        `type`="text"),
                    list(
                        `name`="pathway_id", 
                        `title`="Pathway ID", 
                        `type`="text"),
                    list(
                        `name`="n_biomarkers", 
                        `title`="N Biomarkers", 
                        `type`="integer"),
                    list(
                        `name`="enrichment_score", 
                        `title`="Enrichment Score", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="fdr_p_value", 
                        `title`="FDR P-value", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="biomarkers_in_pathway", 
                        `title`="Biomarkers in Pathway", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="cross_validation_results",
                title="Cross-Validation Results",
                columns=list(
                    list(
                        `name`="fold", 
                        `title`="Fold", 
                        `type`="integer"),
                    list(
                        `name`="n_biomarkers_selected", 
                        `title`="N Biomarkers", 
                        `type`="integer"),
                    list(
                        `name`="auc_roc", 
                        `title`="AUC-ROC", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="accuracy", 
                        `title`="Accuracy", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="sensitivity", 
                        `title`="Sensitivity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="specificity", 
                        `title`="Specificity", 
                        `type`="number", 
                        `format`="zto"),
                    list(
                        `name`="biomarker_overlap", 
                        `title`="Biomarker Overlap %", 
                        `type`="number", 
                        `format`="pc"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="generalizability_assessment",
                title="Generalizability Assessment",
                visible="(biomarker_generalizability)",
                columns=list(
                    list(
                        `name`="assessment_aspect", 
                        `title`="Assessment Aspect", 
                        `type`="text"),
                    list(
                        `name`="current_study", 
                        `title`="Current Study", 
                        `type`="text"),
                    list(
                        `name`="generalization_potential", 
                        `title`="Generalization Potential", 
                        `type`="text"),
                    list(
                        `name`="risk_factors", 
                        `title`="Risk Factors", 
                        `type`="text"),
                    list(
                        `name`="recommendations", 
                        `title`="Recommendations", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="clinical_interpretation",
                title="Clinical Interpretation and Recommendations",
                visible="(biomarker_report)",
                columns=list(
                    list(
                        `name`="aspect", 
                        `title`="Clinical Aspect", 
                        `type`="text"),
                    list(
                        `name`="finding", 
                        `title`="Key Finding", 
                        `type`="text"),
                    list(
                        `name`="clinical_significance", 
                        `title`="Clinical Significance", 
                        `type`="text"),
                    list(
                        `name`="actionability", 
                        `title`="Actionability", 
                        `type`="text"),
                    list(
                        `name`="recommendation", 
                        `title`="Recommendation", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="regulatory_summary",
                title="Regulatory Documentation Summary",
                visible="(regulatory_documentation)",
                columns=list(
                    list(
                        `name`="regulatory_aspect", 
                        `title`="Regulatory Aspect", 
                        `type`="text"),
                    list(
                        `name`="compliance_status", 
                        `title`="Compliance Status", 
                        `type`="text"),
                    list(
                        `name`="documentation", 
                        `title`="Documentation", 
                        `type`="text"),
                    list(
                        `name`="recommendations", 
                        `title`="Recommendations", 
                        `type`="text"))))
            self$add(jmvcore::Image$new(
                options=options,
                name="biomarker_importance_plot",
                title="Biomarker Importance Plot",
                renderFun=".plot_biomarker_importance",
                clearWith=list(
                    "biomarker_vars",
                    "discovery_method")))
            self$add(jmvcore::Image$new(
                options=options,
                name="roc_comparison_plot",
                title="ROC Curve Comparison",
                renderFun=".plot_roc_comparison",
                clearWith=list(
                    "outcome_var",
                    "signature_development")))
            self$add(jmvcore::Image$new(
                options=options,
                name="shap_summary_plot",
                title="SHAP Summary Plot",
                visible="(shap_analysis)",
                renderFun=".plot_shap_summary",
                clearWith=list(
                    "biomarker_vars",
                    "shap_analysis")))
            self$add(jmvcore::Image$new(
                options=options,
                name="shap_dependence_plot",
                title="SHAP Partial Dependence Plots",
                visible="(partial_dependence && shap_analysis)",
                renderFun=".plot_shap_dependence",
                clearWith=list(
                    "biomarker_vars",
                    "partial_dependence")))
            self$add(jmvcore::Image$new(
                options=options,
                name="stability_plot",
                title="Biomarker Selection Stability",
                visible="(stability_analysis)",
                renderFun=".plot_stability",
                clearWith=list(
                    "validation_method",
                    "stability_analysis")))
            self$add(jmvcore::Image$new(
                options=options,
                name="biomarker_correlation_plot",
                title="Selected Biomarker Correlation Matrix",
                renderFun=".plot_biomarker_correlation",
                clearWith=list(
                    "biomarker_vars")))
            self$add(jmvcore::Image$new(
                options=options,
                name="risk_stratification_plot",
                title="Risk Stratification Visualization",
                visible="(risk_stratification)",
                renderFun=".plot_risk_stratification",
                clearWith=list(
                    "risk_stratification",
                    "outcome_var")))
            self$add(jmvcore::Image$new(
                options=options,
                name="decision_curve_plot",
                title="Decision Curve Analysis Plot",
                visible="(decision_curve_analysis)",
                renderFun=".plot_decision_curve",
                clearWith=list(
                    "decision_curve_analysis")))
            self$add(jmvcore::Image$new(
                options=options,
                name="pathway_network_plot",
                title="Pathway Network Visualization",
                visible="(pathway_analysis && biomarker_networks)",
                renderFun=".plot_pathway_network",
                clearWith=list(
                    "pathway_analysis",
                    "biomarker_networks")))
            self$add(jmvcore::Image$new(
                options=options,
                name="biomarker_distribution_plot",
                title="Biomarker Distribution by Outcome",
                renderFun=".plot_biomarker_distribution",
                clearWith=list(
                    "biomarker_vars",
                    "outcome_var")))
            self$add(jmvcore::Image$new(
                options=options,
                name="nomogram_plot",
                title="Biomarker Nomogram",
                visible="(nomogram_development)",
                renderFun=".plot_nomogram",
                clearWith=list(
                    "nomogram_development",
                    "selected_biomarkers")))}))

biomarkerdiscoveryBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "biomarkerdiscoveryBase",
    inherit = jmvcore::Analysis,
    public = list(
        initialize = function(options, data=NULL, datasetId="", analysisId="", revision=0) {
            super$initialize(
                package = "ClinicoPath",
                name = "biomarkerdiscovery",
                version = c(1,0,0),
                options = options,
                results = biomarkerdiscoveryResults$new(options=options),
                data = data,
                datasetId = datasetId,
                analysisId = analysisId,
                revision = revision,
                pause = NULL,
                completeWhenFilled = FALSE,
                requiresMissings = FALSE,
                weightsSupport = 'auto')
        }))

#' Biomarker Discovery Platform with ML Interpretability
#'
#' Comprehensive biomarker discovery and validation platform using machine 
#' learning
#' algorithms with interpretability analysis. Includes feature selection, 
#' biomarker
#' ranking, pathway analysis, and clinical validation metrics. Designed for 
#' omics
#' data analysis with regulatory compliance features for biomarker development
#' and clinical translation.
#' 
#'
#' @examples
#' data('biomarker_data')
#'
#' biomarkerdiscovery(
#'     data = biomarker_data,
#'     outcome_var = "response",
#'     biomarker_vars = c("gene1", "gene2", "protein1"),
#'     discovery_method = "elastic_net",
#'     validation_method = "bootstrap",
#'     interpretability = TRUE
#' )
#'
#' @param data the data as a data frame
#' @param outcome_var Primary outcome variable for biomarker discovery
#' @param biomarker_vars Potential biomarker variables (genes, proteins,
#'   metabolites, etc.)
#' @param clinical_vars Clinical variables to include in the analysis (age,
#'   stage, etc.)
#' @param batch_var Batch or study identifier for batch effect correction
#' @param patient_id Patient identifier for tracking
#' @param time_var Time to event variable for survival biomarker analysis
#' @param event_var Event indicator for survival biomarker analysis
#' @param discovery_method Method for biomarker discovery and selection
#' @param outcome_type Type of outcome variable
#' @param data_type Type of biomarker data being analyzed
#' @param data_preprocessing Perform data preprocessing and normalization
#' @param normalization_method Method for data normalization
#' @param batch_correction Perform batch effect correction
#' @param batch_method Method for batch effect correction
#' @param filter_low_variance Remove features with low variance
#' @param variance_threshold Minimum variance threshold for feature filtering
#' @param feature_selection_method Method for initial feature selection
#' @param n_features_select Maximum number of features to select for analysis
#' @param fdr_threshold False discovery rate threshold for multiple testing
#'   correction
#' @param correlation_threshold Correlation threshold for removing highly
#'   correlated features
#' @param validation_method Method for model validation
#' @param train_proportion Proportion of data for training (70\% = 0.7)
#' @param hyperparameter_tuning Perform hyperparameter optimization
#' @param n_bootstrap_samples Number of bootstrap samples for confidence
#'   intervals
#' @param random_seed Random seed for reproducibility
#' @param biomarker_ranking Rank biomarkers by importance and clinical
#'   relevance
#' @param stability_analysis Assess biomarker selection stability across
#'   resampling
#' @param clinical_performance Calculate clinical performance metrics for
#'   biomarkers
#' @param signature_development Develop multi-biomarker signatures
#' @param pathway_analysis Perform pathway enrichment analysis for discovered
#'   biomarkers
#' @param interpretability Generate interpretability analysis using SHAP/LIME
#' @param shap_analysis Generate SHAP values for biomarker explanation
#' @param lime_analysis Generate LIME explanations for individual predictions
#' @param feature_interaction Analyze interactions between biomarkers
#' @param partial_dependence Generate partial dependence plots for key
#'   biomarkers
#' @param biomarker_networks Analyze biomarker co-expression and interaction
#'   networks
#' @param cutpoint_optimization Find optimal cutpoints for biomarker
#'   classification
#' @param risk_stratification Create risk stratification based on biomarker
#'   signatures
#' @param nomogram_development Develop clinical nomogram incorporating
#'   biomarkers
#' @param decision_curve_analysis Assess clinical utility using decision curve
#'   analysis
#' @param external_validation Prepare biomarkers for external validation
#' @param biomarker_generalizability Assess biomarker generalizability across
#'   populations
#' @param robustness_testing Test biomarker robustness to data perturbations
#' @param quality_control Comprehensive quality control for biomarker data
#' @param outlier_detection Detect and handle outliers in biomarker data
#' @param missing_data_analysis Analyze and handle missing biomarker data
#' @param detailed_results Include comprehensive analysis results
#' @param biomarker_report Generate comprehensive biomarker discovery report
#' @param export_biomarkers Export list of discovered biomarkers
#' @param save_signature Save trained biomarker signature model
#' @param regulatory_documentation Include documentation for regulatory
#'   submission
#' @return A results object containing:
#' \tabular{llllll}{
#'   \code{results$discovery_overview} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$data_summary} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$quality_control_summary} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$outlier_analysis} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$feature_selection_summary} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$selected_biomarkers} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$discovery_performance} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$signature_performance} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$biomarker_ranking} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$stability_analysis_results} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$shap_biomarker_importance} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$biomarker_interactions} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$optimal_cutpoints} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$risk_stratification_results} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$decision_curve_results} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$pathway_enrichment} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$cross_validation_results} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$generalizability_assessment} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$clinical_interpretation} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$regulatory_summary} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$biomarker_importance_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$roc_comparison_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$shap_summary_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$shap_dependence_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$stability_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$biomarker_correlation_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$risk_stratification_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$decision_curve_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$pathway_network_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$biomarker_distribution_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$nomogram_plot} \tab \tab \tab \tab \tab an image \cr
#' }
#'
#' Tables can be converted to data frames with \code{asDF} or \code{\link{as.data.frame}}. For example:
#'
#' \code{results$discovery_overview$asDF}
#'
#' \code{as.data.frame(results$discovery_overview)}
#'
#' @export
biomarkerdiscovery <- function(
    data,
    outcome_var,
    biomarker_vars,
    clinical_vars,
    batch_var,
    patient_id,
    time_var,
    event_var,
    discovery_method = "elastic_net",
    outcome_type = "binary",
    data_type = "genomics",
    data_preprocessing = TRUE,
    normalization_method = "z_score",
    batch_correction = FALSE,
    batch_method = "combat",
    filter_low_variance = TRUE,
    variance_threshold = 0.1,
    feature_selection_method = "univariate_stats",
    n_features_select = 50,
    fdr_threshold = 0.05,
    correlation_threshold = 0.9,
    validation_method = "cv_10fold",
    train_proportion = 0.7,
    hyperparameter_tuning = TRUE,
    n_bootstrap_samples = 1000,
    random_seed = 42,
    biomarker_ranking = TRUE,
    stability_analysis = TRUE,
    clinical_performance = TRUE,
    signature_development = TRUE,
    pathway_analysis = FALSE,
    interpretability = TRUE,
    shap_analysis = TRUE,
    lime_analysis = FALSE,
    feature_interaction = TRUE,
    partial_dependence = TRUE,
    biomarker_networks = FALSE,
    cutpoint_optimization = TRUE,
    risk_stratification = TRUE,
    nomogram_development = TRUE,
    decision_curve_analysis = TRUE,
    external_validation = TRUE,
    biomarker_generalizability = TRUE,
    robustness_testing = TRUE,
    quality_control = TRUE,
    outlier_detection = TRUE,
    missing_data_analysis = TRUE,
    detailed_results = TRUE,
    biomarker_report = TRUE,
    export_biomarkers = FALSE,
    save_signature = FALSE,
    regulatory_documentation = TRUE) {

    if ( ! requireNamespace("jmvcore", quietly=TRUE))
        stop("biomarkerdiscovery requires jmvcore to be installed (restart may be required)")

    if ( ! missing(outcome_var)) outcome_var <- jmvcore::resolveQuo(jmvcore::enquo(outcome_var))
    if ( ! missing(biomarker_vars)) biomarker_vars <- jmvcore::resolveQuo(jmvcore::enquo(biomarker_vars))
    if ( ! missing(clinical_vars)) clinical_vars <- jmvcore::resolveQuo(jmvcore::enquo(clinical_vars))
    if ( ! missing(batch_var)) batch_var <- jmvcore::resolveQuo(jmvcore::enquo(batch_var))
    if ( ! missing(patient_id)) patient_id <- jmvcore::resolveQuo(jmvcore::enquo(patient_id))
    if ( ! missing(time_var)) time_var <- jmvcore::resolveQuo(jmvcore::enquo(time_var))
    if ( ! missing(event_var)) event_var <- jmvcore::resolveQuo(jmvcore::enquo(event_var))
    if (missing(data))
        data <- jmvcore::marshalData(
            parent.frame(),
            `if`( ! missing(outcome_var), outcome_var, NULL),
            `if`( ! missing(biomarker_vars), biomarker_vars, NULL),
            `if`( ! missing(clinical_vars), clinical_vars, NULL),
            `if`( ! missing(batch_var), batch_var, NULL),
            `if`( ! missing(patient_id), patient_id, NULL),
            `if`( ! missing(time_var), time_var, NULL),
            `if`( ! missing(event_var), event_var, NULL))

    for (v in batch_var) if (v %in% names(data)) data[[v]] <- as.factor(data[[v]])
    for (v in event_var) if (v %in% names(data)) data[[v]] <- as.factor(data[[v]])

    options <- biomarkerdiscoveryOptions$new(
        outcome_var = outcome_var,
        biomarker_vars = biomarker_vars,
        clinical_vars = clinical_vars,
        batch_var = batch_var,
        patient_id = patient_id,
        time_var = time_var,
        event_var = event_var,
        discovery_method = discovery_method,
        outcome_type = outcome_type,
        data_type = data_type,
        data_preprocessing = data_preprocessing,
        normalization_method = normalization_method,
        batch_correction = batch_correction,
        batch_method = batch_method,
        filter_low_variance = filter_low_variance,
        variance_threshold = variance_threshold,
        feature_selection_method = feature_selection_method,
        n_features_select = n_features_select,
        fdr_threshold = fdr_threshold,
        correlation_threshold = correlation_threshold,
        validation_method = validation_method,
        train_proportion = train_proportion,
        hyperparameter_tuning = hyperparameter_tuning,
        n_bootstrap_samples = n_bootstrap_samples,
        random_seed = random_seed,
        biomarker_ranking = biomarker_ranking,
        stability_analysis = stability_analysis,
        clinical_performance = clinical_performance,
        signature_development = signature_development,
        pathway_analysis = pathway_analysis,
        interpretability = interpretability,
        shap_analysis = shap_analysis,
        lime_analysis = lime_analysis,
        feature_interaction = feature_interaction,
        partial_dependence = partial_dependence,
        biomarker_networks = biomarker_networks,
        cutpoint_optimization = cutpoint_optimization,
        risk_stratification = risk_stratification,
        nomogram_development = nomogram_development,
        decision_curve_analysis = decision_curve_analysis,
        external_validation = external_validation,
        biomarker_generalizability = biomarker_generalizability,
        robustness_testing = robustness_testing,
        quality_control = quality_control,
        outlier_detection = outlier_detection,
        missing_data_analysis = missing_data_analysis,
        detailed_results = detailed_results,
        biomarker_report = biomarker_report,
        export_biomarkers = export_biomarkers,
        save_signature = save_signature,
        regulatory_documentation = regulatory_documentation)

    analysis <- biomarkerdiscoveryClass$new(
        options = options,
        data = data)

    analysis$run()

    analysis$results
}

