
# This file is automatically generated, you probably don't want to edit this

highdimcoxOptions <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "highdimcoxOptions",
    inherit = jmvcore::Options,
    public = list(
        initialize = function(
            elapsedtime = NULL,
            outcome = NULL,
            predictors = NULL,
            outcomeLevel = "1",
            regularization_method = "elastic_net",
            alpha_value = 0.5,
            cv_method = "cv_1se",
            cv_folds = 10,
            variable_selection = "none",
            stability_selection = FALSE,
            bootstrap_iterations = 500,
            stability_threshold = 0.8,
            show_regularization_path = TRUE,
            show_cv_plot = TRUE,
            show_variable_importance = TRUE,
            show_coefficients_table = TRUE,
            show_model_diagnostics = TRUE,
            showSummaries = FALSE,
            showExplanations = FALSE, ...) {

            super$initialize(
                package="ClinicoPath",
                name="highdimcox",
                requiresData=TRUE,
                ...)

            private$..elapsedtime <- jmvcore::OptionVariable$new(
                "elapsedtime",
                elapsedtime,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..outcome <- jmvcore::OptionVariable$new(
                "outcome",
                outcome,
                suggested=list(
                    "nominal",
                    "ordinal"),
                permitted=list(
                    "factor",
                    "numeric"))
            private$..predictors <- jmvcore::OptionVariables$new(
                "predictors",
                predictors,
                suggested=list(
                    "continuous",
                    "nominal",
                    "ordinal"),
                permitted=list(
                    "numeric",
                    "factor"))
            private$..outcomeLevel <- jmvcore::OptionString$new(
                "outcomeLevel",
                outcomeLevel,
                default="1")
            private$..regularization_method <- jmvcore::OptionList$new(
                "regularization_method",
                regularization_method,
                options=list(
                    "lasso",
                    "ridge",
                    "elastic_net",
                    "adaptive_lasso"),
                default="elastic_net")
            private$..alpha_value <- jmvcore::OptionNumber$new(
                "alpha_value",
                alpha_value,
                min=0,
                max=1,
                default=0.5)
            private$..cv_method <- jmvcore::OptionList$new(
                "cv_method",
                cv_method,
                options=list(
                    "cv_glmnet",
                    "cv_1se",
                    "cv_min"),
                default="cv_1se")
            private$..cv_folds <- jmvcore::OptionInteger$new(
                "cv_folds",
                cv_folds,
                min=3,
                max=20,
                default=10)
            private$..variable_selection <- jmvcore::OptionList$new(
                "variable_selection",
                variable_selection,
                options=list(
                    "none",
                    "stepwise",
                    "best_subset",
                    "forward_selection"),
                default="none")
            private$..stability_selection <- jmvcore::OptionBool$new(
                "stability_selection",
                stability_selection,
                default=FALSE)
            private$..bootstrap_iterations <- jmvcore::OptionInteger$new(
                "bootstrap_iterations",
                bootstrap_iterations,
                min=100,
                max=1000,
                default=500)
            private$..stability_threshold <- jmvcore::OptionNumber$new(
                "stability_threshold",
                stability_threshold,
                min=0.5,
                max=1,
                default=0.8)
            private$..show_regularization_path <- jmvcore::OptionBool$new(
                "show_regularization_path",
                show_regularization_path,
                default=TRUE)
            private$..show_cv_plot <- jmvcore::OptionBool$new(
                "show_cv_plot",
                show_cv_plot,
                default=TRUE)
            private$..show_variable_importance <- jmvcore::OptionBool$new(
                "show_variable_importance",
                show_variable_importance,
                default=TRUE)
            private$..show_coefficients_table <- jmvcore::OptionBool$new(
                "show_coefficients_table",
                show_coefficients_table,
                default=TRUE)
            private$..show_model_diagnostics <- jmvcore::OptionBool$new(
                "show_model_diagnostics",
                show_model_diagnostics,
                default=TRUE)
            private$..showSummaries <- jmvcore::OptionBool$new(
                "showSummaries",
                showSummaries,
                default=FALSE)
            private$..showExplanations <- jmvcore::OptionBool$new(
                "showExplanations",
                showExplanations,
                default=FALSE)

            self$.addOption(private$..elapsedtime)
            self$.addOption(private$..outcome)
            self$.addOption(private$..predictors)
            self$.addOption(private$..outcomeLevel)
            self$.addOption(private$..regularization_method)
            self$.addOption(private$..alpha_value)
            self$.addOption(private$..cv_method)
            self$.addOption(private$..cv_folds)
            self$.addOption(private$..variable_selection)
            self$.addOption(private$..stability_selection)
            self$.addOption(private$..bootstrap_iterations)
            self$.addOption(private$..stability_threshold)
            self$.addOption(private$..show_regularization_path)
            self$.addOption(private$..show_cv_plot)
            self$.addOption(private$..show_variable_importance)
            self$.addOption(private$..show_coefficients_table)
            self$.addOption(private$..show_model_diagnostics)
            self$.addOption(private$..showSummaries)
            self$.addOption(private$..showExplanations)
        }),
    active = list(
        elapsedtime = function() private$..elapsedtime$value,
        outcome = function() private$..outcome$value,
        predictors = function() private$..predictors$value,
        outcomeLevel = function() private$..outcomeLevel$value,
        regularization_method = function() private$..regularization_method$value,
        alpha_value = function() private$..alpha_value$value,
        cv_method = function() private$..cv_method$value,
        cv_folds = function() private$..cv_folds$value,
        variable_selection = function() private$..variable_selection$value,
        stability_selection = function() private$..stability_selection$value,
        bootstrap_iterations = function() private$..bootstrap_iterations$value,
        stability_threshold = function() private$..stability_threshold$value,
        show_regularization_path = function() private$..show_regularization_path$value,
        show_cv_plot = function() private$..show_cv_plot$value,
        show_variable_importance = function() private$..show_variable_importance$value,
        show_coefficients_table = function() private$..show_coefficients_table$value,
        show_model_diagnostics = function() private$..show_model_diagnostics$value,
        showSummaries = function() private$..showSummaries$value,
        showExplanations = function() private$..showExplanations$value),
    private = list(
        ..elapsedtime = NA,
        ..outcome = NA,
        ..predictors = NA,
        ..outcomeLevel = NA,
        ..regularization_method = NA,
        ..alpha_value = NA,
        ..cv_method = NA,
        ..cv_folds = NA,
        ..variable_selection = NA,
        ..stability_selection = NA,
        ..bootstrap_iterations = NA,
        ..stability_threshold = NA,
        ..show_regularization_path = NA,
        ..show_cv_plot = NA,
        ..show_variable_importance = NA,
        ..show_coefficients_table = NA,
        ..show_model_diagnostics = NA,
        ..showSummaries = NA,
        ..showExplanations = NA)
)

highdimcoxResults <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "highdimcoxResults",
    inherit = jmvcore::Group,
    active = list(
        todo = function() private$.items[["todo"]],
        modelSummary = function() private$.items[["modelSummary"]],
        selectedVariables = function() private$.items[["selectedVariables"]],
        regularizationMetrics = function() private$.items[["regularizationMetrics"]],
        stabilityResults = function() private$.items[["stabilityResults"]],
        dimensionalityReduction = function() private$.items[["dimensionalityReduction"]],
        regularizationPath = function() private$.items[["regularizationPath"]],
        cvPlot = function() private$.items[["cvPlot"]],
        variableImportance = function() private$.items[["variableImportance"]],
        modelDiagnostics = function() private$.items[["modelDiagnostics"]],
        stabilityPlot = function() private$.items[["stabilityPlot"]],
        analysisSummary = function() private$.items[["analysisSummary"]],
        methodExplanation = function() private$.items[["methodExplanation"]]),
    private = list(),
    public=list(
        initialize=function(options) {
            super$initialize(
                options=options,
                name="",
                title="High-Dimensional Cox Regression")
            self$add(jmvcore::Html$new(
                options=options,
                name="todo",
                title="Analysis",
                visible=TRUE))
            self$add(jmvcore::Html$new(
                options=options,
                name="modelSummary",
                title="Model Summary",
                visible="(show_coefficients_table)"))
            self$add(jmvcore::Table$new(
                options=options,
                name="selectedVariables",
                title="Selected Variables",
                visible="(show_coefficients_table)",
                columns=list(
                    list(
                        `name`="variable", 
                        `title`="Variable", 
                        `type`="text"),
                    list(
                        `name`="coefficient", 
                        `title`="\u03B2", 
                        `type`="number"),
                    list(
                        `name`="hazard_ratio", 
                        `title`="HR", 
                        `type`="number"),
                    list(
                        `name`="importance_score", 
                        `title`="Importance", 
                        `type`="number"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="regularizationMetrics",
                title="Regularization Metrics",
                visible=TRUE,
                columns=list(
                    list(
                        `name`="metric", 
                        `title`="Metric", 
                        `type`="text"),
                    list(
                        `name`="value", 
                        `title`="Value", 
                        `type`="text"),
                    list(
                        `name`="interpretation", 
                        `title`="Interpretation", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="stabilityResults",
                title="Stability Selection Results",
                visible="(stability_selection)",
                columns=list(
                    list(
                        `name`="variable", 
                        `title`="Variable", 
                        `type`="text"),
                    list(
                        `name`="selection_probability", 
                        `title`="Selection Probability", 
                        `type`="number"),
                    list(
                        `name`="stable", 
                        `title`="Stable Selection", 
                        `type`="text"),
                    list(
                        `name`="importance_rank", 
                        `title`="Importance Rank", 
                        `type`="integer"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="dimensionalityReduction",
                title="Dimensionality Summary",
                visible=TRUE,
                columns=list(
                    list(
                        `name`="stage", 
                        `title`="Analysis Stage", 
                        `type`="text"),
                    list(
                        `name`="variables_input", 
                        `title`="Input Variables", 
                        `type`="integer"),
                    list(
                        `name`="variables_selected", 
                        `title`="Selected Variables", 
                        `type`="integer"),
                    list(
                        `name`="reduction_ratio", 
                        `title`="Reduction Ratio", 
                        `type`="number"))))
            self$add(jmvcore::Image$new(
                options=options,
                name="regularizationPath",
                title="Regularization Path",
                visible="(show_regularization_path)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="cvPlot",
                title="Cross-Validation Plot",
                visible="(show_cv_plot)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="variableImportance",
                title="Variable Importance Plot",
                visible="(show_variable_importance)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="modelDiagnostics",
                title="Model Diagnostics",
                visible="(show_model_diagnostics)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="stabilityPlot",
                title="Stability Selection Plot",
                visible="(stability_selection)"))
            self$add(jmvcore::Html$new(
                options=options,
                name="analysisSummary",
                title="Analysis Summary",
                visible="(showSummaries)"))
            self$add(jmvcore::Html$new(
                options=options,
                name="methodExplanation",
                title="Methodology",
                visible="(showExplanations)"))}))

highdimcoxBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "highdimcoxBase",
    inherit = jmvcore::Analysis,
    public = list(
        initialize = function(options, data=NULL, datasetId="", analysisId="", revision=0) {
            super$initialize(
                package = "ClinicoPath",
                name = "highdimcox",
                version = c(0,0,1),
                options = options,
                results = highdimcoxResults$new(options=options),
                data = data,
                datasetId = datasetId,
                analysisId = analysisId,
                revision = revision,
                pause = NULL,
                completeWhenFilled = FALSE,
                requiresMissings = FALSE,
                weightsSupport = 'auto')
        }))

#' High-Dimensional Cox Regression
#'
#' Performs high-dimensional Cox proportional hazards regression using 
#' advanced regularization methods (LASSO, Ridge, Elastic Net, Adaptive LASSO) 
#' for  survival data with many predictors. This method is essential when the 
#' number of variables is large relative to the number of observations, 
#' enabling variable selection and preventing overfitting in genomic, 
#' proteomic, or other high-dimensional clinical datasets.
#'
#' @examples
#' # Example 1: LASSO regularization for gene expression data
#' library(survival)
#' library(glmnet)
#'
#' highdimcox(
#'     data = genomic_survival_data,
#'     elapsedtime = "time",
#'     outcome = "status",
#'     outcomeLevel = "1",
#'     predictors = c("gene1", "gene2", "gene3", "...gene1000"),
#'     regularization_method = "lasso",
#'     cv_folds = 10
#' )
#'
#' # Example 2: Elastic Net with stability selection
#' highdimcox(
#'     data = protein_data,
#'     elapsedtime = "survival_time",
#'     outcome = "event",
#'     outcomeLevel = "1",
#'     predictors = c("protein1", "protein2", "...protein500"),
#'     regularization_method = "elastic_net",
#'     alpha_value = 0.5,
#'     stability_selection = TRUE,
#'     bootstrap_iterations = 500
#' )
#'
#' @param data the data as a data frame
#' @param elapsedtime Time variable for survival analysis
#' @param outcome Event indicator variable
#' @param predictors High-dimensional predictor variables
#' @param outcomeLevel Level of outcome variable indicating event
#' @param regularization_method Regularization method for high-dimensional Cox
#'   regression
#' @param alpha_value Alpha parameter for elastic net (0=ridge, 1=lasso)
#' @param cv_method Cross-validation method for lambda selection
#' @param cv_folds Number of cross-validation folds
#' @param variable_selection Additional variable selection method after
#'   regularization
#' @param stability_selection Perform stability selection for variable
#'   importance
#' @param bootstrap_iterations Number of bootstrap iterations for stability
#'   selection
#' @param stability_threshold Threshold for stability selection
#' @param show_regularization_path Display regularization path plot
#' @param show_cv_plot Display cross-validation error plot
#' @param show_variable_importance Display variable importance plot
#' @param show_coefficients_table Display selected coefficients table
#' @param show_model_diagnostics Display model diagnostic plots
#' @param showSummaries Generate natural language summaries
#' @param showExplanations Show methodology explanations
#' @return A results object containing:
#' \tabular{llllll}{
#'   \code{results$todo} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$modelSummary} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$selectedVariables} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$regularizationMetrics} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$stabilityResults} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$dimensionalityReduction} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$regularizationPath} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$cvPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$variableImportance} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$modelDiagnostics} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$stabilityPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$analysisSummary} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$methodExplanation} \tab \tab \tab \tab \tab a html \cr
#' }
#'
#' Tables can be converted to data frames with \code{asDF} or \code{\link{as.data.frame}}. For example:
#'
#' \code{results$selectedVariables$asDF}
#'
#' \code{as.data.frame(results$selectedVariables)}
#'
#' @export
highdimcox <- function(
    data,
    elapsedtime,
    outcome,
    predictors,
    outcomeLevel = "1",
    regularization_method = "elastic_net",
    alpha_value = 0.5,
    cv_method = "cv_1se",
    cv_folds = 10,
    variable_selection = "none",
    stability_selection = FALSE,
    bootstrap_iterations = 500,
    stability_threshold = 0.8,
    show_regularization_path = TRUE,
    show_cv_plot = TRUE,
    show_variable_importance = TRUE,
    show_coefficients_table = TRUE,
    show_model_diagnostics = TRUE,
    showSummaries = FALSE,
    showExplanations = FALSE) {

    if ( ! requireNamespace("jmvcore", quietly=TRUE))
        stop("highdimcox requires jmvcore to be installed (restart may be required)")

    if ( ! missing(elapsedtime)) elapsedtime <- jmvcore::resolveQuo(jmvcore::enquo(elapsedtime))
    if ( ! missing(outcome)) outcome <- jmvcore::resolveQuo(jmvcore::enquo(outcome))
    if ( ! missing(predictors)) predictors <- jmvcore::resolveQuo(jmvcore::enquo(predictors))
    if (missing(data))
        data <- jmvcore::marshalData(
            parent.frame(),
            `if`( ! missing(elapsedtime), elapsedtime, NULL),
            `if`( ! missing(outcome), outcome, NULL),
            `if`( ! missing(predictors), predictors, NULL))


    options <- highdimcoxOptions$new(
        elapsedtime = elapsedtime,
        outcome = outcome,
        predictors = predictors,
        outcomeLevel = outcomeLevel,
        regularization_method = regularization_method,
        alpha_value = alpha_value,
        cv_method = cv_method,
        cv_folds = cv_folds,
        variable_selection = variable_selection,
        stability_selection = stability_selection,
        bootstrap_iterations = bootstrap_iterations,
        stability_threshold = stability_threshold,
        show_regularization_path = show_regularization_path,
        show_cv_plot = show_cv_plot,
        show_variable_importance = show_variable_importance,
        show_coefficients_table = show_coefficients_table,
        show_model_diagnostics = show_model_diagnostics,
        showSummaries = showSummaries,
        showExplanations = showExplanations)

    analysis <- highdimcoxClass$new(
        options = options,
        data = data)

    analysis$run()

    analysis$results
}

