
# This file is automatically generated, you probably don't want to edit this

mendelianrandomizationOptions <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "mendelianrandomizationOptions",
    inherit = jmvcore::Options,
    public = list(
        initialize = function(
            use_same_dataset = TRUE,
            snp_column = NULL,
            beta_exposure = NULL,
            se_exposure = NULL,
            pval_exposure = NULL,
            beta_outcome = NULL,
            se_outcome = NULL,
            pval_outcome = NULL,
            eaf_column = NULL,
            pval_threshold = 5e-8,
            clump_r2 = 0.001,
            clump_kb = 10000,
            mr_methods = "main_three",
            ivw_model = "random",
            egger_bootstrap = FALSE,
            bootstrap_samples = 1000,
            heterogeneity_test = TRUE,
            pleiotropy_test = TRUE,
            leave_one_out = TRUE,
            single_snp_analysis = FALSE,
            mr_presso = FALSE,
            presso_threshold = 0.05,
            plot_forest = TRUE,
            plot_funnel = TRUE,
            plot_scatter = TRUE,
            plot_loo = TRUE,
            harmonize_alleles = TRUE,
            remove_palindromic = TRUE,
            steiger_filtering = FALSE,
            min_snps = 3,
            conf_level = 0.95,
            random_seed = 42, ...) {

            super$initialize(
                package="ClinicoPath",
                name="mendelianrandomization",
                requiresData=TRUE,
                ...)

            private$..use_same_dataset <- jmvcore::OptionBool$new(
                "use_same_dataset",
                use_same_dataset,
                default=TRUE)
            private$..snp_column <- jmvcore::OptionVariable$new(
                "snp_column",
                snp_column,
                suggested=list(
                    "nominal"),
                permitted=list(
                    "factor",
                    "id"))
            private$..beta_exposure <- jmvcore::OptionVariable$new(
                "beta_exposure",
                beta_exposure,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..se_exposure <- jmvcore::OptionVariable$new(
                "se_exposure",
                se_exposure,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..pval_exposure <- jmvcore::OptionVariable$new(
                "pval_exposure",
                pval_exposure,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"),
                default=NULL)
            private$..beta_outcome <- jmvcore::OptionVariable$new(
                "beta_outcome",
                beta_outcome,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..se_outcome <- jmvcore::OptionVariable$new(
                "se_outcome",
                se_outcome,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..pval_outcome <- jmvcore::OptionVariable$new(
                "pval_outcome",
                pval_outcome,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"),
                default=NULL)
            private$..eaf_column <- jmvcore::OptionVariable$new(
                "eaf_column",
                eaf_column,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"),
                default=NULL)
            private$..pval_threshold <- jmvcore::OptionNumber$new(
                "pval_threshold",
                pval_threshold,
                default=5e-8,
                min=1e-9,
                max=0.01)
            private$..clump_r2 <- jmvcore::OptionNumber$new(
                "clump_r2",
                clump_r2,
                default=0.001,
                min=0.0001,
                max=0.1)
            private$..clump_kb <- jmvcore::OptionInteger$new(
                "clump_kb",
                clump_kb,
                default=10000,
                min=250,
                max=10000)
            private$..mr_methods <- jmvcore::OptionList$new(
                "mr_methods",
                mr_methods,
                options=list(
                    "all",
                    "ivw",
                    "egger",
                    "weighted_median",
                    "ivw_egger",
                    "main_three"),
                default="main_three")
            private$..ivw_model <- jmvcore::OptionList$new(
                "ivw_model",
                ivw_model,
                options=list(
                    "fixed",
                    "random",
                    "multiplicative"),
                default="random")
            private$..egger_bootstrap <- jmvcore::OptionBool$new(
                "egger_bootstrap",
                egger_bootstrap,
                default=FALSE)
            private$..bootstrap_samples <- jmvcore::OptionInteger$new(
                "bootstrap_samples",
                bootstrap_samples,
                default=1000,
                min=100,
                max=10000)
            private$..heterogeneity_test <- jmvcore::OptionBool$new(
                "heterogeneity_test",
                heterogeneity_test,
                default=TRUE)
            private$..pleiotropy_test <- jmvcore::OptionBool$new(
                "pleiotropy_test",
                pleiotropy_test,
                default=TRUE)
            private$..leave_one_out <- jmvcore::OptionBool$new(
                "leave_one_out",
                leave_one_out,
                default=TRUE)
            private$..single_snp_analysis <- jmvcore::OptionBool$new(
                "single_snp_analysis",
                single_snp_analysis,
                default=FALSE)
            private$..mr_presso <- jmvcore::OptionBool$new(
                "mr_presso",
                mr_presso,
                default=FALSE)
            private$..presso_threshold <- jmvcore::OptionNumber$new(
                "presso_threshold",
                presso_threshold,
                default=0.05,
                min=0.001,
                max=0.1)
            private$..plot_forest <- jmvcore::OptionBool$new(
                "plot_forest",
                plot_forest,
                default=TRUE)
            private$..plot_funnel <- jmvcore::OptionBool$new(
                "plot_funnel",
                plot_funnel,
                default=TRUE)
            private$..plot_scatter <- jmvcore::OptionBool$new(
                "plot_scatter",
                plot_scatter,
                default=TRUE)
            private$..plot_loo <- jmvcore::OptionBool$new(
                "plot_loo",
                plot_loo,
                default=TRUE)
            private$..harmonize_alleles <- jmvcore::OptionBool$new(
                "harmonize_alleles",
                harmonize_alleles,
                default=TRUE)
            private$..remove_palindromic <- jmvcore::OptionBool$new(
                "remove_palindromic",
                remove_palindromic,
                default=TRUE)
            private$..steiger_filtering <- jmvcore::OptionBool$new(
                "steiger_filtering",
                steiger_filtering,
                default=FALSE)
            private$..min_snps <- jmvcore::OptionInteger$new(
                "min_snps",
                min_snps,
                default=3,
                min=1,
                max=10)
            private$..conf_level <- jmvcore::OptionNumber$new(
                "conf_level",
                conf_level,
                default=0.95,
                min=0.8,
                max=0.99)
            private$..random_seed <- jmvcore::OptionInteger$new(
                "random_seed",
                random_seed,
                default=42,
                min=1,
                max=999999)

            self$.addOption(private$..use_same_dataset)
            self$.addOption(private$..snp_column)
            self$.addOption(private$..beta_exposure)
            self$.addOption(private$..se_exposure)
            self$.addOption(private$..pval_exposure)
            self$.addOption(private$..beta_outcome)
            self$.addOption(private$..se_outcome)
            self$.addOption(private$..pval_outcome)
            self$.addOption(private$..eaf_column)
            self$.addOption(private$..pval_threshold)
            self$.addOption(private$..clump_r2)
            self$.addOption(private$..clump_kb)
            self$.addOption(private$..mr_methods)
            self$.addOption(private$..ivw_model)
            self$.addOption(private$..egger_bootstrap)
            self$.addOption(private$..bootstrap_samples)
            self$.addOption(private$..heterogeneity_test)
            self$.addOption(private$..pleiotropy_test)
            self$.addOption(private$..leave_one_out)
            self$.addOption(private$..single_snp_analysis)
            self$.addOption(private$..mr_presso)
            self$.addOption(private$..presso_threshold)
            self$.addOption(private$..plot_forest)
            self$.addOption(private$..plot_funnel)
            self$.addOption(private$..plot_scatter)
            self$.addOption(private$..plot_loo)
            self$.addOption(private$..harmonize_alleles)
            self$.addOption(private$..remove_palindromic)
            self$.addOption(private$..steiger_filtering)
            self$.addOption(private$..min_snps)
            self$.addOption(private$..conf_level)
            self$.addOption(private$..random_seed)
        }),
    active = list(
        use_same_dataset = function() private$..use_same_dataset$value,
        snp_column = function() private$..snp_column$value,
        beta_exposure = function() private$..beta_exposure$value,
        se_exposure = function() private$..se_exposure$value,
        pval_exposure = function() private$..pval_exposure$value,
        beta_outcome = function() private$..beta_outcome$value,
        se_outcome = function() private$..se_outcome$value,
        pval_outcome = function() private$..pval_outcome$value,
        eaf_column = function() private$..eaf_column$value,
        pval_threshold = function() private$..pval_threshold$value,
        clump_r2 = function() private$..clump_r2$value,
        clump_kb = function() private$..clump_kb$value,
        mr_methods = function() private$..mr_methods$value,
        ivw_model = function() private$..ivw_model$value,
        egger_bootstrap = function() private$..egger_bootstrap$value,
        bootstrap_samples = function() private$..bootstrap_samples$value,
        heterogeneity_test = function() private$..heterogeneity_test$value,
        pleiotropy_test = function() private$..pleiotropy_test$value,
        leave_one_out = function() private$..leave_one_out$value,
        single_snp_analysis = function() private$..single_snp_analysis$value,
        mr_presso = function() private$..mr_presso$value,
        presso_threshold = function() private$..presso_threshold$value,
        plot_forest = function() private$..plot_forest$value,
        plot_funnel = function() private$..plot_funnel$value,
        plot_scatter = function() private$..plot_scatter$value,
        plot_loo = function() private$..plot_loo$value,
        harmonize_alleles = function() private$..harmonize_alleles$value,
        remove_palindromic = function() private$..remove_palindromic$value,
        steiger_filtering = function() private$..steiger_filtering$value,
        min_snps = function() private$..min_snps$value,
        conf_level = function() private$..conf_level$value,
        random_seed = function() private$..random_seed$value),
    private = list(
        ..use_same_dataset = NA,
        ..snp_column = NA,
        ..beta_exposure = NA,
        ..se_exposure = NA,
        ..pval_exposure = NA,
        ..beta_outcome = NA,
        ..se_outcome = NA,
        ..pval_outcome = NA,
        ..eaf_column = NA,
        ..pval_threshold = NA,
        ..clump_r2 = NA,
        ..clump_kb = NA,
        ..mr_methods = NA,
        ..ivw_model = NA,
        ..egger_bootstrap = NA,
        ..bootstrap_samples = NA,
        ..heterogeneity_test = NA,
        ..pleiotropy_test = NA,
        ..leave_one_out = NA,
        ..single_snp_analysis = NA,
        ..mr_presso = NA,
        ..presso_threshold = NA,
        ..plot_forest = NA,
        ..plot_funnel = NA,
        ..plot_scatter = NA,
        ..plot_loo = NA,
        ..harmonize_alleles = NA,
        ..remove_palindromic = NA,
        ..steiger_filtering = NA,
        ..min_snps = NA,
        ..conf_level = NA,
        ..random_seed = NA)
)

mendelianrandomizationResults <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "mendelianrandomizationResults",
    inherit = jmvcore::Group,
    active = list(
        instructions = function() private$.items[["instructions"]],
        dataInfo = function() private$.items[["dataInfo"]],
        snpSelection = function() private$.items[["snpSelection"]],
        selectedSNPs = function() private$.items[["selectedSNPs"]],
        mrResults = function() private$.items[["mrResults"]],
        heterogeneityTable = function() private$.items[["heterogeneityTable"]],
        pleiotropyTable = function() private$.items[["pleiotropyTable"]],
        looTable = function() private$.items[["looTable"]],
        pressoResults = function() private$.items[["pressoResults"]],
        forestPlot = function() private$.items[["forestPlot"]],
        funnelPlot = function() private$.items[["funnelPlot"]],
        scatterPlot = function() private$.items[["scatterPlot"]],
        looPlot = function() private$.items[["looPlot"]],
        interpretation = function() private$.items[["interpretation"]],
        assumptions = function() private$.items[["assumptions"]]),
    private = list(),
    public=list(
        initialize=function(options) {
            super$initialize(
                options=options,
                name="",
                title="Mendelian Randomization")
            self$add(jmvcore::Html$new(
                options=options,
                name="instructions",
                title="Instructions",
                visible=TRUE))
            self$add(jmvcore::Preformatted$new(
                options=options,
                name="dataInfo",
                title="Data Information",
                visible=TRUE))
            self$add(jmvcore::Html$new(
                options=options,
                name="snpSelection",
                title="SNP Selection Summary",
                visible=TRUE))
            self$add(jmvcore::Table$new(
                options=options,
                name="selectedSNPs",
                title="Selected Instrumental Variables",
                rows=0,
                columns=list(
                    list(
                        `name`="snp", 
                        `title`="SNP", 
                        `type`="text"),
                    list(
                        `name`="beta_exp", 
                        `title`="\u03B2 (Exposure)", 
                        `type`="number"),
                    list(
                        `name`="se_exp", 
                        `title`="SE (Exposure)", 
                        `type`="number"),
                    list(
                        `name`="pval_exp", 
                        `title`="P-value (Exp)", 
                        `type`="number", 
                        `format`="zto,pvalue"),
                    list(
                        `name`="beta_out", 
                        `title`="\u03B2 (Outcome)", 
                        `type`="number"),
                    list(
                        `name`="se_out", 
                        `title`="SE (Outcome)", 
                        `type`="number"),
                    list(
                        `name`="pval_out", 
                        `title`="P-value (Out)", 
                        `type`="number", 
                        `format`="zto,pvalue"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="mrResults",
                title="Mendelian Randomization Estimates",
                rows=0,
                columns=list(
                    list(
                        `name`="method", 
                        `title`="Method", 
                        `type`="text"),
                    list(
                        `name`="n_snps", 
                        `title`="N SNPs", 
                        `type`="integer"),
                    list(
                        `name`="estimate", 
                        `title`="Estimate", 
                        `type`="number"),
                    list(
                        `name`="se", 
                        `title`="SE", 
                        `type`="number"),
                    list(
                        `name`="ci_lower", 
                        `title`="Lower", 
                        `superTitle`="95% CI", 
                        `type`="number"),
                    list(
                        `name`="ci_upper", 
                        `title`="Upper", 
                        `superTitle`="95% CI", 
                        `type`="number"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto,pvalue"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="heterogeneityTable",
                title="Heterogeneity Statistics",
                rows=0,
                visible="(heterogeneity_test)",
                columns=list(
                    list(
                        `name`="method", 
                        `title`="Method", 
                        `type`="text"),
                    list(
                        `name`="q_statistic", 
                        `title`="Q Statistic", 
                        `type`="number"),
                    list(
                        `name`="q_df", 
                        `title`="df", 
                        `type`="integer"),
                    list(
                        `name`="q_pval", 
                        `title`="Q P-value", 
                        `type`="number", 
                        `format`="zto,pvalue"),
                    list(
                        `name`="i2", 
                        `title`="I\u00B2 (%)", 
                        `type`="number"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="pleiotropyTable",
                title="Horizontal Pleiotropy Test (MR-Egger Intercept)",
                rows=0,
                visible="(pleiotropy_test)",
                columns=list(
                    list(
                        `name`="intercept", 
                        `title`="Intercept", 
                        `type`="number"),
                    list(
                        `name`="se", 
                        `title`="SE", 
                        `type`="number"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number", 
                        `format`="zto,pvalue"),
                    list(
                        `name`="interpretation", 
                        `title`="Interpretation", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="looTable",
                title="Leave-One-Out Analysis",
                rows=0,
                visible="(leave_one_out)",
                columns=list(
                    list(
                        `name`="snp_removed", 
                        `title`="SNP Removed", 
                        `type`="text"),
                    list(
                        `name`="estimate", 
                        `title`="Estimate", 
                        `type`="number"),
                    list(
                        `name`="ci_lower", 
                        `title`="Lower CI", 
                        `type`="number"),
                    list(
                        `name`="ci_upper", 
                        `title`="Upper CI", 
                        `type`="number"))))
            self$add(jmvcore::Html$new(
                options=options,
                name="pressoResults",
                title="MR-PRESSO Outlier Analysis",
                visible="(mr_presso)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="forestPlot",
                title="Forest Plot - MR Estimates",
                width=600,
                height=450,
                renderFun=".plotForest",
                visible="(plot_forest)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="funnelPlot",
                title="Funnel Plot - Asymmetry Assessment",
                width=600,
                height=450,
                renderFun=".plotFunnel",
                visible="(plot_funnel)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="scatterPlot",
                title="Scatter Plot - SNP Effects",
                width=600,
                height=450,
                renderFun=".plotScatter",
                visible="(plot_scatter)"))
            self$add(jmvcore::Image$new(
                options=options,
                name="looPlot",
                title="Leave-One-Out Plot",
                width=600,
                height=450,
                renderFun=".plotLeaveOneOut",
                visible="(plot_loo)"))
            self$add(jmvcore::Html$new(
                options=options,
                name="interpretation",
                title="Interpretation Guide",
                visible=TRUE))
            self$add(jmvcore::Html$new(
                options=options,
                name="assumptions",
                title="MR Assumptions Check",
                visible=TRUE))}))

mendelianrandomizationBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "mendelianrandomizationBase",
    inherit = jmvcore::Analysis,
    public = list(
        initialize = function(options, data=NULL, datasetId="", analysisId="", revision=0) {
            super$initialize(
                package = "ClinicoPath",
                name = "mendelianrandomization",
                version = c(0,0,32),
                options = options,
                results = mendelianrandomizationResults$new(options=options),
                data = data,
                datasetId = datasetId,
                analysisId = analysisId,
                revision = revision,
                pause = NULL,
                completeWhenFilled = FALSE,
                requiresMissings = FALSE,
                weightsSupport = 'auto')
        }))

#' Mendelian Randomization
#'
#' Mendelian Randomization (MR) uses genetic variants as instrumental
#' variables to infer causal relationships between exposures and outcomes.
#' 
#' This module supports GWAS summary statistics (NOT individual-level data)
#' and provides multiple MR methods with comprehensive sensitivity analysis.
#' 
#' Applications: Genetic epidemiology, drug target validation, causal
#' inference in observational studies.
#' 
#'
#' @examples
#' # Example: Analyze effect of BMI on coronary disease
#' mendelianrandomization(
#'     exposure_data = bmi_gwas,
#'     outcome_data = cad_gwas,
#'     snp_column = 'rsID',
#'     beta_exposure = 'beta_bmi',
#'     se_exposure = 'se_bmi',
#'     beta_outcome = 'beta_cad',
#'     se_outcome = 'se_cad',
#'     mr_methods = c('IVW', 'MR-Egger', 'Weighted Median')
#' )
#'
#' @param exposure_data GWAS summary statistics for exposure variable
#' @param outcome_data GWAS summary statistics for outcome variable (optional
#'   if same dataset)
#' @param use_same_dataset Whether exposure and outcome are in the same
#'   dataset
#' @param snp_column Column containing SNP identifiers (e.g., rsID)
#' @param beta_exposure Effect size (beta) for exposure
#' @param se_exposure Standard error for exposure beta
#' @param pval_exposure P-value for exposure association
#' @param beta_outcome Effect size (beta) for outcome
#' @param se_outcome Standard error for outcome beta
#' @param pval_outcome P-value for outcome association
#' @param eaf_column Effect allele frequency (optional)
#' @param pval_threshold Genome-wide significance threshold (default 5e-8)
#' @param clump_r2 RÂ² threshold for LD clumping
#' @param clump_kb Distance window for LD clumping in kilobases
#' @param mr_methods Which MR methods to perform
#' @param ivw_model Model specification for IVW method
#' @param egger_bootstrap Use bootstrap for MR-Egger confidence intervals
#' @param bootstrap_samples Number of bootstrap samples
#' @param heterogeneity_test Perform Cochran Q heterogeneity test
#' @param pleiotropy_test Test for horizontal pleiotropy (MR-Egger intercept)
#' @param leave_one_out Perform leave-one-out sensitivity analysis
#' @param single_snp_analysis Analyze each SNP individually
#' @param mr_presso Detect and correct for outliers using MR-PRESSO
#' @param presso_threshold P-value threshold for MR-PRESSO outlier detection
#' @param plot_forest Generate forest plot of MR estimates
#' @param plot_funnel Generate funnel plot for asymmetry assessment
#' @param plot_scatter Generate scatter plot with method lines
#' @param plot_loo Plot leave-one-out analysis results
#' @param harmonize_alleles Harmonize exposure and outcome effect alleles
#' @param remove_palindromic Remove palindromic SNPs with ambiguous strand
#' @param steiger_filtering Filter SNPs using Steiger test for directionality
#' @param min_snps Minimum number of SNPs required for analysis
#' @param conf_level Confidence level for intervals
#' @param random_seed Random seed for reproducibility
#' @return A results object containing:
#' \tabular{llllll}{
#'   \code{results$instructions} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$dataInfo} \tab \tab \tab \tab \tab a preformatted \cr
#'   \code{results$snpSelection} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$selectedSNPs} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$mrResults} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$heterogeneityTable} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$pleiotropyTable} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$looTable} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$pressoResults} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$forestPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$funnelPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$scatterPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$looPlot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$interpretation} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$assumptions} \tab \tab \tab \tab \tab a html \cr
#' }
#'
#' Tables can be converted to data frames with \code{asDF} or \code{\link{as.data.frame}}. For example:
#'
#' \code{results$selectedSNPs$asDF}
#'
#' \code{as.data.frame(results$selectedSNPs)}
#'
#' @export
mendelianrandomization <- function(
    exposure_data,
    outcome_data,
    use_same_dataset = TRUE,
    snp_column,
    beta_exposure,
    se_exposure,
    pval_exposure = NULL,
    beta_outcome,
    se_outcome,
    pval_outcome = NULL,
    eaf_column = NULL,
    pval_threshold = 5e-8,
    clump_r2 = 0.001,
    clump_kb = 10000,
    mr_methods = "main_three",
    ivw_model = "random",
    egger_bootstrap = FALSE,
    bootstrap_samples = 1000,
    heterogeneity_test = TRUE,
    pleiotropy_test = TRUE,
    leave_one_out = TRUE,
    single_snp_analysis = FALSE,
    mr_presso = FALSE,
    presso_threshold = 0.05,
    plot_forest = TRUE,
    plot_funnel = TRUE,
    plot_scatter = TRUE,
    plot_loo = TRUE,
    harmonize_alleles = TRUE,
    remove_palindromic = TRUE,
    steiger_filtering = FALSE,
    min_snps = 3,
    conf_level = 0.95,
    random_seed = 42) {

    if ( ! requireNamespace("jmvcore", quietly=TRUE))
        stop("mendelianrandomization requires jmvcore to be installed (restart may be required)")

    if ( ! missing(snp_column)) snp_column <- jmvcore::resolveQuo(jmvcore::enquo(snp_column))
    if ( ! missing(beta_exposure)) beta_exposure <- jmvcore::resolveQuo(jmvcore::enquo(beta_exposure))
    if ( ! missing(se_exposure)) se_exposure <- jmvcore::resolveQuo(jmvcore::enquo(se_exposure))
    if ( ! missing(pval_exposure)) pval_exposure <- jmvcore::resolveQuo(jmvcore::enquo(pval_exposure))
    if ( ! missing(beta_outcome)) beta_outcome <- jmvcore::resolveQuo(jmvcore::enquo(beta_outcome))
    if ( ! missing(se_outcome)) se_outcome <- jmvcore::resolveQuo(jmvcore::enquo(se_outcome))
    if ( ! missing(pval_outcome)) pval_outcome <- jmvcore::resolveQuo(jmvcore::enquo(pval_outcome))
    if ( ! missing(eaf_column)) eaf_column <- jmvcore::resolveQuo(jmvcore::enquo(eaf_column))
    if (missing(data))
        data <- jmvcore::marshalData(
            parent.frame(),
            `if`( ! missing(snp_column), snp_column, NULL),
            `if`( ! missing(beta_exposure), beta_exposure, NULL),
            `if`( ! missing(se_exposure), se_exposure, NULL),
            `if`( ! missing(pval_exposure), pval_exposure, NULL),
            `if`( ! missing(beta_outcome), beta_outcome, NULL),
            `if`( ! missing(se_outcome), se_outcome, NULL),
            `if`( ! missing(pval_outcome), pval_outcome, NULL),
            `if`( ! missing(eaf_column), eaf_column, NULL))


    options <- mendelianrandomizationOptions$new(
        use_same_dataset = use_same_dataset,
        snp_column = snp_column,
        beta_exposure = beta_exposure,
        se_exposure = se_exposure,
        pval_exposure = pval_exposure,
        beta_outcome = beta_outcome,
        se_outcome = se_outcome,
        pval_outcome = pval_outcome,
        eaf_column = eaf_column,
        pval_threshold = pval_threshold,
        clump_r2 = clump_r2,
        clump_kb = clump_kb,
        mr_methods = mr_methods,
        ivw_model = ivw_model,
        egger_bootstrap = egger_bootstrap,
        bootstrap_samples = bootstrap_samples,
        heterogeneity_test = heterogeneity_test,
        pleiotropy_test = pleiotropy_test,
        leave_one_out = leave_one_out,
        single_snp_analysis = single_snp_analysis,
        mr_presso = mr_presso,
        presso_threshold = presso_threshold,
        plot_forest = plot_forest,
        plot_funnel = plot_funnel,
        plot_scatter = plot_scatter,
        plot_loo = plot_loo,
        harmonize_alleles = harmonize_alleles,
        remove_palindromic = remove_palindromic,
        steiger_filtering = steiger_filtering,
        min_snps = min_snps,
        conf_level = conf_level,
        random_seed = random_seed)

    analysis <- mendelianrandomizationClass$new(
        options = options,
        data = data)

    analysis$run()

    analysis$results
}

