
# This file is automatically generated, you probably don't want to edit this

superpcOptions <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "superpcOptions",
    inherit = jmvcore::Options,
    public = list(
        initialize = function(
            time = NULL,
            event = NULL,
            features = NULL,
            threshold = 0.1,
            n_components = 5,
            cv_folds = 10,
            standardize = TRUE,
            screening_method = "univariate_cox",
            pca_method = "standard",
            validation_method = "cv",
            plot_screening = TRUE,
            plot_pca = TRUE,
            plot_survival = TRUE,
            export_components = FALSE, ...) {

            super$initialize(
                package="ClinicoPath",
                name="superpc",
                requiresData=TRUE,
                ...)

            private$..time <- jmvcore::OptionVariable$new(
                "time",
                time,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..event <- jmvcore::OptionVariable$new(
                "event",
                event,
                suggested=list(
                    "nominal"),
                permitted=list(
                    "factor"))
            private$..features <- jmvcore::OptionVariables$new(
                "features",
                features,
                suggested=list(
                    "continuous"),
                permitted=list(
                    "numeric"))
            private$..threshold <- jmvcore::OptionNumber$new(
                "threshold",
                threshold,
                default=0.1,
                min=0.01,
                max=0.5)
            private$..n_components <- jmvcore::OptionNumber$new(
                "n_components",
                n_components,
                default=5,
                min=1,
                max=50)
            private$..cv_folds <- jmvcore::OptionNumber$new(
                "cv_folds",
                cv_folds,
                default=10,
                min=3,
                max=20)
            private$..standardize <- jmvcore::OptionBool$new(
                "standardize",
                standardize,
                default=TRUE)
            private$..screening_method <- jmvcore::OptionList$new(
                "screening_method",
                screening_method,
                options=list(
                    "univariate_cox",
                    "correlation",
                    "t_test"),
                default="univariate_cox")
            private$..pca_method <- jmvcore::OptionList$new(
                "pca_method",
                pca_method,
                options=list(
                    "standard",
                    "sparse",
                    "robust"),
                default="standard")
            private$..validation_method <- jmvcore::OptionList$new(
                "validation_method",
                validation_method,
                options=list(
                    "cv",
                    "bootstrap",
                    "split_sample"),
                default="cv")
            private$..plot_screening <- jmvcore::OptionBool$new(
                "plot_screening",
                plot_screening,
                default=TRUE)
            private$..plot_pca <- jmvcore::OptionBool$new(
                "plot_pca",
                plot_pca,
                default=TRUE)
            private$..plot_survival <- jmvcore::OptionBool$new(
                "plot_survival",
                plot_survival,
                default=TRUE)
            private$..export_components <- jmvcore::OptionBool$new(
                "export_components",
                export_components,
                default=FALSE)

            self$.addOption(private$..time)
            self$.addOption(private$..event)
            self$.addOption(private$..features)
            self$.addOption(private$..threshold)
            self$.addOption(private$..n_components)
            self$.addOption(private$..cv_folds)
            self$.addOption(private$..standardize)
            self$.addOption(private$..screening_method)
            self$.addOption(private$..pca_method)
            self$.addOption(private$..validation_method)
            self$.addOption(private$..plot_screening)
            self$.addOption(private$..plot_pca)
            self$.addOption(private$..plot_survival)
            self$.addOption(private$..export_components)
        }),
    active = list(
        time = function() private$..time$value,
        event = function() private$..event$value,
        features = function() private$..features$value,
        threshold = function() private$..threshold$value,
        n_components = function() private$..n_components$value,
        cv_folds = function() private$..cv_folds$value,
        standardize = function() private$..standardize$value,
        screening_method = function() private$..screening_method$value,
        pca_method = function() private$..pca_method$value,
        validation_method = function() private$..validation_method$value,
        plot_screening = function() private$..plot_screening$value,
        plot_pca = function() private$..plot_pca$value,
        plot_survival = function() private$..plot_survival$value,
        export_components = function() private$..export_components$value),
    private = list(
        ..time = NA,
        ..event = NA,
        ..features = NA,
        ..threshold = NA,
        ..n_components = NA,
        ..cv_folds = NA,
        ..standardize = NA,
        ..screening_method = NA,
        ..pca_method = NA,
        ..validation_method = NA,
        ..plot_screening = NA,
        ..plot_pca = NA,
        ..plot_survival = NA,
        ..export_components = NA)
)

superpcResults <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "superpcResults",
    inherit = jmvcore::Group,
    active = list(
        instructions = function() private$.items[["instructions"]],
        analysis_summary = function() private$.items[["analysis_summary"]],
        feature_screening = function() private$.items[["feature_screening"]],
        principal_components = function() private$.items[["principal_components"]],
        component_loadings = function() private$.items[["component_loadings"]],
        model_performance = function() private$.items[["model_performance"]],
        risk_groups = function() private$.items[["risk_groups"]],
        cross_validation_results = function() private$.items[["cross_validation_results"]],
        feature_screening_plot = function() private$.items[["feature_screening_plot"]],
        pca_biplot = function() private$.items[["pca_biplot"]],
        variance_explained_plot = function() private$.items[["variance_explained_plot"]],
        survival_curves_plot = function() private$.items[["survival_curves_plot"]],
        component_heatmap = function() private$.items[["component_heatmap"]],
        clinical_interpretation = function() private$.items[["clinical_interpretation"]]),
    private = list(),
    public=list(
        initialize=function(options) {
            super$initialize(
                options=options,
                name="",
                title="Supervised Principal Components Results",
                refs=list(
                    "ClinicoPathJamoviModule"))
            self$add(jmvcore::Html$new(
                options=options,
                name="instructions",
                title="Instructions",
                visible=TRUE))
            self$add(jmvcore::Table$new(
                options=options,
                name="analysis_summary",
                title="Analysis Summary",
                rows=1,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="total_features", 
                        `title`="Total Features", 
                        `type`="integer"),
                    list(
                        `name`="selected_features", 
                        `title`="Selected Features", 
                        `type`="integer"),
                    list(
                        `name`="n_components", 
                        `title`="Components Used", 
                        `type`="integer"),
                    list(
                        `name`="variance_explained", 
                        `title`="Variance Explained (%)", 
                        `type`="number"),
                    list(
                        `name`="c_index", 
                        `title`="C-index", 
                        `type`="number"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="feature_screening",
                title="Feature Screening Results",
                rows=0,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="feature", 
                        `title`="Feature", 
                        `type`="text"),
                    list(
                        `name`="univariate_hr", 
                        `title`="Univariate HR", 
                        `type`="number"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number"),
                    list(
                        `name`="selected", 
                        `title`="Selected", 
                        `type`="text"),
                    list(
                        `name`="rank", 
                        `title`="Rank", 
                        `type`="integer"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="principal_components",
                title="Principal Components",
                rows=0,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="component", 
                        `title`="Component", 
                        `type`="text"),
                    list(
                        `name`="variance_explained", 
                        `title`="Variance Explained (%)", 
                        `type`="number"),
                    list(
                        `name`="cumulative_variance", 
                        `title`="Cumulative Variance (%)", 
                        `type`="number"),
                    list(
                        `name`="cox_coefficient", 
                        `title`="Cox Coefficient", 
                        `type`="number"),
                    list(
                        `name`="hazard_ratio", 
                        `title`="Hazard Ratio", 
                        `type`="number"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="component_loadings",
                title="Component Loadings",
                rows=0,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="feature", 
                        `title`="Feature", 
                        `type`="text"),
                    list(
                        `name`="pc1_loading", 
                        `title`="PC1 Loading", 
                        `type`="number"),
                    list(
                        `name`="pc2_loading", 
                        `title`="PC2 Loading", 
                        `type`="number"),
                    list(
                        `name`="pc3_loading", 
                        `title`="PC3 Loading", 
                        `type`="number"),
                    list(
                        `name`="max_loading", 
                        `title`="Max |Loading|", 
                        `type`="number"),
                    list(
                        `name`="dominant_pc", 
                        `title`="Dominant PC", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="model_performance",
                title="Model Performance",
                rows=0,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="metric", 
                        `title`="Metric", 
                        `type`="text"),
                    list(
                        `name`="training", 
                        `title`="Training", 
                        `type`="number"),
                    list(
                        `name`="validation", 
                        `title`="Cross-Validation", 
                        `type`="number"),
                    list(
                        `name`="confidence_interval", 
                        `title`="95% CI", 
                        `type`="text"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="risk_groups",
                title="Risk Group Analysis",
                rows=0,
                visible=TRUE,
                columns=list(
                    list(
                        `name`="risk_group", 
                        `title`="Risk Group", 
                        `type`="text"),
                    list(
                        `name`="n_patients", 
                        `title`="N Patients", 
                        `type`="integer"),
                    list(
                        `name`="events", 
                        `title`="Events", 
                        `type`="integer"),
                    list(
                        `name`="median_survival", 
                        `title`="Median Survival", 
                        `type`="number"),
                    list(
                        `name`="hazard_ratio", 
                        `title`="Hazard Ratio", 
                        `type`="number"),
                    list(
                        `name`="p_value", 
                        `title`="P-value", 
                        `type`="number"))))
            self$add(jmvcore::Table$new(
                options=options,
                name="cross_validation_results",
                title="Cross-Validation Results",
                rows=0,
                visible="(validation_method:cv)",
                columns=list(
                    list(
                        `name`="fold", 
                        `title`="Fold", 
                        `type`="integer"),
                    list(
                        `name`="c_index", 
                        `title`="C-index", 
                        `type`="number"),
                    list(
                        `name`="selected_features", 
                        `title`="Selected Features", 
                        `type`="integer"),
                    list(
                        `name`="components_used", 
                        `title`="Components Used", 
                        `type`="integer"))))
            self$add(jmvcore::Image$new(
                options=options,
                name="feature_screening_plot",
                title="Feature Screening Results",
                width=600,
                height=450,
                visible="(plot_screening)",
                renderFun=".plot_feature_screening"))
            self$add(jmvcore::Image$new(
                options=options,
                name="pca_biplot",
                title="PCA Biplot",
                width=600,
                height=450,
                visible="(plot_pca)",
                renderFun=".plot_pca_biplot"))
            self$add(jmvcore::Image$new(
                options=options,
                name="variance_explained_plot",
                title="Variance Explained",
                width=600,
                height=450,
                visible="(plot_pca)",
                renderFun=".plot_variance_explained"))
            self$add(jmvcore::Image$new(
                options=options,
                name="survival_curves_plot",
                title="Survival Curves by Risk Groups",
                width=600,
                height=450,
                visible="(plot_survival)",
                renderFun=".plot_survival_curves"))
            self$add(jmvcore::Image$new(
                options=options,
                name="component_heatmap",
                title="Component Loadings Heatmap",
                width=600,
                height=450,
                visible="(plot_pca)",
                renderFun=".plot_component_heatmap"))
            self$add(jmvcore::Html$new(
                options=options,
                name="clinical_interpretation",
                title="Clinical Interpretation and Biological Insights",
                visible=TRUE))}))

superpcBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
    "superpcBase",
    inherit = jmvcore::Analysis,
    public = list(
        initialize = function(options, data=NULL, datasetId="", analysisId="", revision=0) {
            super$initialize(
                package = "ClinicoPath",
                name = "superpc",
                version = c(0,0,31),
                options = options,
                results = superpcResults$new(options=options),
                data = data,
                datasetId = datasetId,
                analysisId = analysisId,
                revision = revision,
                pause = NULL,
                completeWhenFilled = FALSE,
                requiresMissings = FALSE,
                weightsSupport = 'auto')
        }))

#' Supervised Principal Components Cox Regression
#'
#' Supervised Principal Components (SuperPC) for Cox regression with 
#' high-dimensional data.
#' This method first screens features using univariate Cox regression, then 
#' applies PCA
#' to selected features, and finally fits a Cox model using the principal 
#' components.
#' Particularly effective for genomics data where many features are 
#' correlated.
#' 
#'
#' @examples
#' superpc(
#'     data = data,
#'     time = "time",
#'     event = "event",
#'     features = c("gene1", "gene2", "gene3"),
#'     threshold = 0.1,
#'     n_components = 5
#' )
#'
#' @param data the data as a data frame
#' @param time survival time variable
#' @param event event indicator (1=event, 0=censored)
#' @param features high-dimensional feature variables (e.g., gene expression)
#' @param threshold p-value threshold for initial feature screening
#' @param n_components number of principal components to extract
#' @param cv_folds number of folds for cross-validation
#' @param standardize standardize features before analysis
#' @param screening_method method for initial feature screening
#' @param pca_method principal component analysis method
#' @param validation_method model validation approach
#' @param plot_screening display feature screening results
#' @param plot_pca display principal component analysis plots
#' @param plot_survival display survival curves by risk groups
#' @param export_components export principal component scores to data
#' @return A results object containing:
#' \tabular{llllll}{
#'   \code{results$instructions} \tab \tab \tab \tab \tab a html \cr
#'   \code{results$analysis_summary} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$feature_screening} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$principal_components} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$component_loadings} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$model_performance} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$risk_groups} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$cross_validation_results} \tab \tab \tab \tab \tab a table \cr
#'   \code{results$feature_screening_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$pca_biplot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$variance_explained_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$survival_curves_plot} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$component_heatmap} \tab \tab \tab \tab \tab an image \cr
#'   \code{results$clinical_interpretation} \tab \tab \tab \tab \tab a html \cr
#' }
#'
#' Tables can be converted to data frames with \code{asDF} or \code{\link{as.data.frame}}. For example:
#'
#' \code{results$analysis_summary$asDF}
#'
#' \code{as.data.frame(results$analysis_summary)}
#'
#' @export
superpc <- function(
    data,
    time,
    event,
    features,
    threshold = 0.1,
    n_components = 5,
    cv_folds = 10,
    standardize = TRUE,
    screening_method = "univariate_cox",
    pca_method = "standard",
    validation_method = "cv",
    plot_screening = TRUE,
    plot_pca = TRUE,
    plot_survival = TRUE,
    export_components = FALSE) {

    if ( ! requireNamespace("jmvcore", quietly=TRUE))
        stop("superpc requires jmvcore to be installed (restart may be required)")

    if ( ! missing(time)) time <- jmvcore::resolveQuo(jmvcore::enquo(time))
    if ( ! missing(event)) event <- jmvcore::resolveQuo(jmvcore::enquo(event))
    if ( ! missing(features)) features <- jmvcore::resolveQuo(jmvcore::enquo(features))
    if (missing(data))
        data <- jmvcore::marshalData(
            parent.frame(),
            `if`( ! missing(time), time, NULL),
            `if`( ! missing(event), event, NULL),
            `if`( ! missing(features), features, NULL))

    for (v in event) if (v %in% names(data)) data[[v]] <- as.factor(data[[v]])

    options <- superpcOptions$new(
        time = time,
        event = event,
        features = features,
        threshold = threshold,
        n_components = n_components,
        cv_folds = cv_folds,
        standardize = standardize,
        screening_method = screening_method,
        pca_method = pca_method,
        validation_method = validation_method,
        plot_screening = plot_screening,
        plot_pca = plot_pca,
        plot_survival = plot_survival,
        export_components = export_components)

    analysis <- superpcClass$new(
        options = options,
        data = data)

    analysis$run()

    analysis$results
}

